\documentclass[fleqn,10pt]{wlpeerj}
\title{Large, structurally-curated alignment and phylogeny of Vertebrate biogenic amine receptors}

\author[1,2,3]{Stephanie J. Spielman}
\author[4,5]{Ahmad R. Sedaghat}
\author[1,2,3]{Keerthana Kumar}
\author[1,2,3]{Claus O. Wilke}
\affil[1]{Department of Integrative Biology, The University of Texas at Austin, Austin, U.S.A.}
\affil[2]{Institute of Cellular and Molecular Biology, The University of Texas at Austin, Austin, U.S.A.}
\affil[3]{Center for Computational Biology and Bioinformatics, The University of Texas at Austin, Austin, U.S.A.}
\affil[4]{Department of Otolaryngologyâ€“Head and Neck Surgery, Massachusetts Eye and Ear Infirmary, Boston, Massachusetts, U.S.A.}
\affil[5]{Department of Otology and Laryngology, Harvard Medical School, Boston, Massachusetts, U.S.A.}

\keywords{biogenic amine receptors, phylogenetics, multiple sequence alignment, G-protein coupled receptors}


\begin{abstract}


\end{abstract}

\begin{document}

\flushbottom
\maketitle
\thispagestyle{empty}


\section*{Introduction}

Biogenic amines, such as the molecules serotonin and dopamine, play critical roles in virtually all Metazoa taxa, exerting significant influence on both behavior and physiology particularly within the central nervous system. In Vertebrate species, their actions are mediated primarily through the biogenic amine receptor family, which includes dopamine (DRD), histamine (HRH), trace (TAAR), adrenergic (ADR), muscarinic cholinergic (mAChR), and most serotonin receptors (5HTR).  Biogenic amine receptors belong to the broad family of G protein-coupled receptors (GPCR), one of the largest and most diverse receptor families in eukaryota, and indeed Metazoa. Indeed, due to the extensive diversity of biological functions they direct and the ongoing expansion of their ligand repertoire, GPCRs are considered one of the most evolutionarily innovative and successful gene families \citep{BockaertPin1999,Lagerstrom2008}.

Biogenic amine receptors fall in to the Rhodopsin-like (family ``A'') GPCR family, which emerged as a distinct GPCR clade with the origins of the Opisthokont (Fungi and Metazoa) lineage \citep{Krishnan2012}. Subsequently, the Rhodopsin-like family underwent substantial expansion in Metazoa, most notably in Vertebrate lineages \citep{Rompleretal2007,Staubert2013}. Like all GPCRs, these receptors are highly structurally conserved, containing seven transmembrane (TM) domains separated by three extracellular (ECL) and three intracellular (ICL) loops with an N-outside C-inside orientation, and they propagate intracellular signaling through a G protein-mediated pathway. Biogenic amine receptors are frequent targets for a wide array of pharmaceuticals aimed to treat diseases such as schizophrenia, migraines, hypertension, allergies and asthma, and stomach ulcers \citep{Schoneberg2004,Eversetal2005,Masonetal2012}.

In spite of these receptors' biological and clinical importance, studies on their evolution are relatively limited. Evolutionary studies which have focused on a biogenic amine receptors have predominantly been limited to single receptor subtypes, namely TAAR \citep{Gloriametal2005,Lindemann2005,Hashiguchi2007}, DRD \citep{Callieretal2003,Yamamotoetal2013}, and 5HTR \cite{Anbazhagan2010}. Moreover, many of these studies, and indeed larger-scale studies on the general evolution of the broad Rhodopsin-like GPCR family, have examined very narrow species distributions, for instance specifically teleosts \citep{Gloriametal2005}, primates \citep{Anbazhagan2010}, humans and mice \citep{Vassilatis2003,KakaralaJamil2014}, or even strictly humans \cite{Fredrikssonetal2003}. Therefore, studies which account for the full breadth of known bioamine receptor sequences as well as their broad species distributions 

This dearth of biogenic amine receptor-specific evolutionary understanding is underscored by the difficulties in establishing a robust multiple sequence alignment (MSA). Commonly used to identify conserved sequence motifs and functionally important residues as well as investigate the sequences' evolutionary history, MSAs provide the foundation for nearly all comparative sequence analyses. As MSA errors readily introduce bias these downstream analyses \citep{Ogden2006, Wong2008, Jordan2012}, it is crucial to ensure accuracy in the alignment to the extent possible. For GPCR sequences, in particular, any alignment should recapitulate their canonical 7TM structure, which a naive alignment of sequences cannot necessarily accomplish. While there are certain MSA software platforms which explicitly incorporate structural information into the alignment algorithm (e.g.\ 3DCoffe \citep{3dcoffee} and PROMALS3D \citep{promals3d}), these programs are fairly computationally-intensive and thus unwieldy (ill-suited?) for large-scale (over 1000 sequences) applications. Moreover, such programs require the use of a single crystal structure to guide sequence alignment. While all GPCRs have the same conserved 7TM domains, different GPCRs, and indeed the biogenic amine receptors, feature a wide variety of ICL and ECL sizes. For instance, the ECL3 lengths for human HRH1 and DRD3, respectively, are roughly 27 and 117, and their respective ICL3 lengths are 68 and 14 (as predicted by GPCRHMM \citep{Wistrand2006}). Thus, aligning with a single crystral structure will not effectively represent the domain variability across biogenic amine receptor subtypes. A desirable alignment strategy would ensure that all sequences are anchored by their conserved 7TM domains without inappropriately constraining the heterogeneous ECL and ICL domains.

We therefore have adopted a novel iterative alignment strategy to create a large (3064 sequences), structurally-curated MSA of vertebrate biogenic amine receptors. In order to ensure proper structural alignment, we employed the software GPCRHMM \citep{Wistrand2006}, which uses a hidden markov model approach to assign each residue in a given GPCR sequence to its respective domain, either extracellular, transmembrane, or intracellular. Previously validated with real GPCR crystal structures \citep{SpielmanWilke2013}, GPCRHMM relies on the remarkably-conserved GPCR 7TM structure to predict GPCR domains with exceptional accuracy and without the use of a crystal structure. Our alignment strategy therefore ensured that all predicted 7TM domains aligned appropriately across all sequences. Importantly, this strategy did not require any manual or visual data inspection, thus avoiding any confounding subjectivity in MSA processing. 

We present this biogenic amine receptor MSA as a resource for any group interested in studying the dynamic evolutionary processes, structural, and/or functional constraints operating on this exceptionally important GPCR clade. Moreover, to demonstrate the utility of our alignment, we constructed a phylogeny of our receptors with this alignment as well as a naive MSA which did not undergo this iterative process. We found that our structurally-curated MSA offered dramatic improvements over a structurally-naive MSA. Our structurally-aware MSA and corresponding phylogeny therefore represent the most comprehensive and curated vertebrate biogenic amine receptor dataset to date. Moreover, using this phylogeny, we are able to discern the relationships among receptor types with a far increased sensitivity than previous analyses. Even so, we are unable to fully resolve the deeper nodes in the phylogeny, revealing the capabilities and limitations of current-day phylogenetic methods.

\section*{Results}

\subsection*{Constructing a structurally-curated MSA of biogenic amine receptors}
We collected all sequences using PSI-BLAST and filtered data to ensure a high-quality data set (see \emph{Methods} for details). We additionally retained only sequences which the program GPCRHMM unequivocally classified as a true GPCR sequence, leaving a total dataset of 3464 receptor sequences to align. We then built a structurally-curated MSA from these 3464 receptor protein sequences according to the iterative strategy outlined in Figure~\ref{flowchart}. The resulting alignment, obtained after four iterations of this algorithm, contained a total of 3039 sequences. However, according to this algorithm, sequences in which up to 5\% of positions do not match the consensus domain are retained. Therefore, certain residues which lie at domain boundaries may be improperly aligned with regards to structure. Thus, we created a modified version of this structural alignment in which all residues whose GPCRHMM-assigned domain did not agree with its respective column's consensus domain were masked. In total, 2.69\% of all MSA positions were masked, and therefore treated as missing characters. Moreover, the final MSA contains strictly sequences within the  \emph{Euteleostomi} clade of jawed vertebrates (Figure~\ref{taxa_dist}). The MSA is particularly enriched for sequences from Eutherian (placental mammal) species, likely due to the stringent filters we applied to sequence collection which favored fully-sequenced genomes.  

\subsection*{Structurally-aware approach strongly improves phylogenetic inference}

To demonstrate the utility of our biogenic amine receptor MSA, we constructed five distinct maximum likelihood (ML) phylogenies, as detailed in Table~\ref{tab:phylo_AIC}, in RAxML. First, we built a phylogeny using an MSA, again built with MAFFT \citep{mafftv7}, comprised of the full set of 3464 receptors; in other words, this MSA was not subjected to the iterative process shown in Figure~\ref{flowchart}. As this MSA was not structurally-curated, we refer to it as the ``naive'' MSA. We additionally constructed two phylogenies each from the structural unmasked and masked MSAs. Previous work has shown that both structural and functional constraints impose differing selection pressures in TM vs.\ extramembrane (EM), which consists of both extracellular and intracellular, domains, additionally producing distinct amino-acid frequency distributions in each domain class \cite{Tourasse2000,Stevens2001,Julenius2006,Oberai2009,SpielmanWilke2013,FranzosaXueXia2013}. As our structurally-curated MSAs allow us to precisely identify each MSA column as either TM or EM, we can conduct far more rigorous phylogentic inference using a partition analysis. Therefore, for each of our two structurally-curated MSAs (masked and unmasked), we inferred two ML phylogenies: one with two partitions representing TM and EM columns and one with a single partition for the entire alignment. Importantly, such a partitioned analysis would not be possible without a structurally-curated MSA, as we present here.

The structurally-curated masked MSA generated a far superior phylogeny, based on AIC scores, relative to all other MSAs (Table~\ref{tab:phylo_AIC}), highlighting the benefits of structurally-aware analyses. That the masked structural MSA yielded phylogenies with substantially better fits than unmasked MSA underscores that any structurally-aware study must be undertaken cautiously. While partitioning the MSA based on structural domains was clearly benefically, ensuring that these domains were accurately assigned was critical. Having even a few TM residues in a column assigned to the EM partition, or vice versa, strongly hindered phylogenetic fit, explaining the improved performance of the masked structural MSA.



\subsection*{Structurally-aware phylogeny reveals unknown biogenic amine receptor relationships and clades}

Our resulting phylogeny, shown in Figure~\ref{phylogeny}, represents the most comprehensive and curated vertebrate biogenic amine receptor phylogeny to date. This tree broadly captures many known features of biogenic amine receptor evolution, in particular that these receptors do not cluster based on ligand-binding but rather have undergone extensive functional convergent evolution. Indeed, our phylogeny reveals that only two ligand-based receptor classes, mAChR and TAAR, are truly monophyletic. Using this phylogeny, we were able to reclassify several misannotated sequences (Table S1) as well as uncover an entirely unknown clade of biogenic amine receptors. This unknown clade, sister to HRH2, contains strictly avian sequences as well as one \emph{Xenopus tropicalis} sequence. Two evolutionary scenarios may explain this taxonomic distribution: either this clade emerged after the divergence of teleosts but was secondarily lost in Reptilia and Mammalia, or this clade represents an avian-specific diversification which the \emph{Xenopus tropicalis} sequence resembles only convergently. Interestingly, the vast majority of sequences in this clade were annotated in NCBI as either octopamine or No9-like receptors, both of which are insect-specific biogenic amine receptors that do not occur in vertebrate taxa. This sequence misannotation reveals an intriguing case of convergent evolution and suggests the hypothesis that these receptors may interact with atypical ligands in vertebrate lineages.

Our phylogeny features remarkably high bootstrap support for each distinct clade of receptor subtypes. We additionally find very strong support for three deeper nodes in the phylogeny that reveal the relationships among distinct receptor subtypes. The first contains the three clades HRH1, mAChR, and HRH-3,4, the second contains the clades 5HTR-1, 5HTR-5, and 5HTR-7, and finally the third contains the 5HTR-4 and TAAR clades. Previous studies have yielded conflicting phylogenetic placements for the 5HTR-7 clade; some have argued that 5HTR-7 is phylogenetically distinct from all other 5HTR sequences \citep{KakaralaJamil2014}, while others have found that evidence for a single clade containing 5HTR-5,7 as a sister taxa to a clade containing ADRA1 sequences \citep{Fredrikssonetal2003}. Alternatively, we find moderate-to-strong support for the 5HTR-7 clade having originated before subsequent diversification into 5HTR-5 and 5HTR-1, and we find full support showing that ADRA1 sequences form an entire distinct monophyletic group outside all other vertebrate biogenic amine receptors. In addition, as previously mentioned, our phylogeny reveals that HRH-3,4 is actually single monophyletic group. Moreover, the HRH-4 clade contains strictly mammalians sequences, including monotreme sequences, wheres the HRH-3 sequences are broadly distributed among vertebrate taxa. We therefore hypothesize that HRH-4 arose from an HRH-3 duplication concurrent with mammalian origins.

Of particular interest in our phylogeny are the unique evolutionary patterns revealed within the TAAR clade. While TAAR sequences do cluster together, the relationships among TAAR subtypes are highly dynamic, reflecting the extensive expansion and contraction events characterizing this receptor family's evolution \citep{Lindemann2005,Hashiguchi2007,Staubert2010,Staubert2013}. Figure~\ref{taar_tree} displays the subtree containing specifically TAAR sequences and in fact differs somewhat from previously proposed TAAR trees \citep{Lindemann2005, Hashiguchi2007}. In particular, the presence of several lineage-specific subclades as well as unresolved subclades generate novel hypotheses regarding TAAR subtype origins. While TAAR-2,3,4 form a well-resolved monophyly, sister to a clade containing subtypes TAAR-5,6,7,8,9, the latter clade is less straight-forward to interpret. Indeed, the TAAR subtypes -6, -7, and -9 do not constitute distinct monophyletic groups, suggesting either poor NCBI sequence annotation or rampant diversification within this subclade. If we assume that these NCBI annotations are reasonably correct, we can deduce that this clade's ancestral sequence was most similar to TAAR-7 and subsequently diversified independently into TAAR-9 and TAAR-6, which in turn gave rise to the monophyletic TAAR-8. Furthermore, lobe-finned (coelacanth) sequences do not clearly cluster with any TAAR subtypes, likely due to this lineage's ancient divergence and unique evolutionary trajectory of this lineage \citep{coelacanth2013}. The phylogenetic distribution of lobe-finned fish sequences may aid future endeavors to tease apart evolutionary origins of certain TAAR subtypes, specifically whether they represent teleost-specific duplications \citep{Gloriametal2005} or whether they represent ancient TAARs that emerged before teleost divergence but were secondarily lost in lobe-finned fish and/or tetrapods.

In addition, a small clade sister to TAAR (labeled in Figure~\ref{phylogeny} and Figure~\ref{taar_tree}as TAAR$\ast$) strictly contains sequences annotated by NCBI as ``5HTR4-like,'' which might suggest that 5HTR-4 is in fact paraphyletic, diversifying gradually before giving rise to TAARs. However, all sequences in TAAR$\ast$ belong taxonomically either to teleost or \emph{Xenopus tropicalis}. Thus, we suspect that these sequences are actually misannotated, and therefore this clade in fact corresponds to the so-called TAAR V cluster identified by \cite{Hashiguchi2007}. Indeed, \cite{Hashiguchi2007} showed that the TAAR V cluster is an outgroup to all other vertebrate TAAR sequences, and moreover that 5HTR-4 is an outgroup to the overall TAAR clade, as our phylogeny similarly shows.


Finally, we emphasize the limits of phylogenetic inference for understanding the complex evolutionary histories of expanding gene families. The majority of the deeper splits in our phylogeny have fairly weak bootstrap support, and thus this phylogeny alone is not sufficient for fully resolving the relationships among biogenic amine receptor classes. Indeed, modern phylogenetic inference methods are ill-suited for deducing such relationships, particularly because MSA gaps are treated simply as missing data. In reality, gaps represent the evolutionary events of insertion and deletions (indels). Unfortunately, current phylogenetic methods focus solely on the substitution process, effectively ignoring that gaps are indeed evolutionary events containing important information and ultimately hindering phylogenetic accuracy \cite{Morrison2008,Loytynoja2008,Warnow2012,Luanetal2013}. Like most protein families, GPCRs do not simply diversify through nucleotide substitutions, as novel GPCRs tend to arise after major indel events in the ICL and ECL domains \citep{BockaertPin1999}. Unfortunately, the evolutionary intermediates connecting sequences have long-since disappeared from genomes, and there is no obvious way to infer the sequences of these ``missing links.'' Thus, additional approaches, notably syntenic analyses \citep{Sundstrom2010,Widmark2011,YegorovGood2012,Hwangetal2013}, combined with the phylogeny presented here should prove useful towards resolving the evolutionary history of vertebrate biogenic amine receptors. 

\section*{Conclusions}


To clearly address origins, this phylogeny and alignment may be combined with a molecular clock or synteny analysis.

Any large-scale study requires a robust sequence alignment to identify protein homology, and then can examine the overlap between receptor-specific trends and broader patterns which fall across all of these things. Furthermore, although biogenic amine receptors are relatively more straightforward to generate pharms for than other GPCRs, their therapeutics still suffer from a lack of complete specificty. Having a thorough understanding of the general patterns and rules governing this clade's sequence evolution, and indeed understanding the permissible amino acids at structurally important positions, may enable substantial progress in homology modeling and pharm development for these important receptors.
Therefore, please feel free to use our alignment for evolutionary analysis of amine receptors, GPCRs, transmembrane domains. Also have great use for identifying certain sequence motifs as well as identifying broad trends in allowed sequence variability, can examine the sequence variability within and among different receptor subtypes to identify unifying or distinguishing patterns or evolutionary principles. May prove useful in extending homology modeling of these receptors as well as phamaceutical development. 



\section*{Methods}

\subsection*{Sequence Collection and Processing}
We collected protein sequences using PSI-BLAST \citep{psiblast}, specifically from the RefSeq ($v2.2.29+$) database \citep{refseq}, for 42 distinct human biogenic amine receptor sequences representing the full range of known receptors in the human genome. To obtain distant yet well-supported orthologs, we ran each PSI-BLAST search for 5 iterations with a e-value cutoff of $10^{-20}$, a sequence identity threshold of 25\%, and a length difference of 50\%. After combining all sequences recovered from the individual PSI-BLAST searches, we discarded duplicate sequences, leaving a total of 4232 PSI-BLAST results. We then filtered this sequence set by removing sequences from non-vertebrate taxa, sequences annotated as low-quality, pseudogene, and/or partial, and sequences which contained more than 1\% ambiguous (i.e.\ B, X, or Z) residues. We additionally removed any sequences that could not be robustly considered GPCRs. We used the program GPCRHMM \citep{Wistrand2006} to determine whether a given sequence was indeed a GPCR, and we discarded sequences which had either a local or global GPCRHMM score less than 10, both extremely conservative thresholds. Thus, while it is possible that some true GPCRs were discarded, these high thresholds for both local and global scores provide a very high confidence that all retained sequences were indeed GPCRs. Together, these filters left a total of 3464 receptor sequences.


\subsection*{Sequence Alignment}
Before aligning sequences, we used the program GPCRHMM \citep{Wistrand2006} to assign each residue in all protein sequences to its respective structural domain (extracellular, transmembrane, or intracellular) using a $0.5$ posterior probability cutoff. As all GPCRs are comprised of precise and highly conserved structural domains, it is critical that any sequence alignment of GPCRs maintains this overall homologous structure. In particular, the 7 TM domains should properly align with one another, across all sequences. Unfortunately, current alignment methods which consider structural information, such as 3DCoffee \citep{3dcoffee} and PROMALS3D \citep{promals3d} are ill-suited for such large datasets. Therefore, we adopted a novel alignment strategy to ensure that the resulting alignment accuractely reflected the highly conserved overarching GPCR structure. We began by assigning, again using GPCRHMM \citep{Wistrand2006}, 

Our alignment strategy, outlined in Figure~\ref{flowchart}, consisted of a series of alignment iterations. Within each iteration, we first obtained an alignment using MAFFT (v7.149b) \cite{mafftv7}. Next, using the GPCRHMM-determined residue domains, we determined the consensus structural domain for each alignment column, and we discarded all sequences in which 5\% of columns did not match this consensus structure. We then realigned the remaining sequences, performing this strategy until no sequences were discarded. The final structurally-curated MSA contained 3039 sequences. In addition to this final alignment, we additionally created a ``masked'' alignment, in which protein residues which did not conform to their respective consensus domains were replaced with a ``?'', thus effectively replacing these positions with an ambiguous character. Therefore, while the final alignment might contain a few improperly aligned residues, this issue is avoided in the masked alignment as each column contains strictly properly-aligned residues.  




\section*{Acknowledgments}

Funded by NIH, army, probably.

\bibliography{bibliography}


\newpage


\section*{Figures and Tables}

\vspace{3cm}

\begin{figure}[htbp]
	\centerline{\includegraphics[width=18cm]{figures/alignment_flowchart.pdf}}
	\caption{\label{flowchart} Iterative alignment strategy to create a structurally-curated multiple sequence alignment. Starred sequence is consensus structure. I=intracellular, M=membrane, E=extracellular. Sequences were removed if $\geq 5\%$ of columns belonged to a different structural domain than the respective consensus domain.}
\end{figure}


\newpage


\begin{figure}[htbp]
	\centerline{\includegraphics[width=7cm]{figures/taxonomic_distribution.pdf}}
	\caption{\label{taxa_dist} Cladogram of the taxonomic distribution of all sequences in the final structurally-curated alignment. All sequences were members of the \emph{Euteleostomi} clade of jawed vertebrates. Numbers in parentheses indicate the total number of biogenic amine receptors in the respective clade.}
\end{figure}

\vspace{3cm}

\begin{table}[htbp]
	\centering
	\begin{tabular}{l c l l c}
		\hline\noalign{\smallskip}
		\multicolumn{1}{c}{Alignment} & \multicolumn{1}{c}{Partitions?} & \multicolumn{1}{c}{$\ln L$} & \multicolumn{1}{c}{k} & \multicolumn{1}{l}{$\Delta$AIC} \\
		\hline\noalign{\smallskip}
		Structural Masked & Yes & -505500.8 & 6115 & 0 \\
		Structural Masked & No & -515991.7 & 6095 & 1752 \\  
		Structural Unmasked & Yes & -515343.6 & 6115 & 19685 \\
		Structural Unmasked & No & -515991.7 & 6095 & 20941 \\ 
		Naive & No &  -589703.7 & 6945 & 170047 \\
		\noalign{\smallskip}\hline\noalign{\smallskip} 
	\end{tabular}
	\caption{\label{tab:phylo_AIC} $\Delta$AIC scores relative to the best performing for phylogenies.  AIC is computed as AIC $= 2(k - \ln L)$, where $k$ is the number of free parameters of the model, and $\ln L$ is the log-likelihood \citep{Akaike1974,BurnhamAnderson2004}.}
\end{table}


\newpage

\begin{figure}[htbp]
	\centerline{\includegraphics[width=18cm]{figures/masked_part_phylogeny.pdf}}
	\caption{\label{phylogeny} Maximum likelihood phylogeny of Vertebrate bioamine receptors built using the masked structural protein MSA alignment in RAxML. Nodes with open circles indicate $\geq 50\%$ bootstrap support, and nodes with closed circles and thick lines indicate $\geq 90\%$ bootstrap support. Bioaminergic receptors are abbreviated as 5HTR, serotonin receptors; TAAR, trace amine-associated receptors; HRH, histamine receptors; mAChr, muscarinic acetylcholine receptors; ADR, adreneric receptors; and DRD, dopamine receptors. The clade labeled ``Unknown'' could not be clearly identified as one of the major receptor types. Likely, this is a herp-specific gene...}
\end{figure}


\newpage

\begin{figure}[htbp]
	\centerline{\includegraphics[width=15cm]{figures/taar_phylogeny.pdf}}
	\caption{\label{taar_tree} Specifically the TAAR clades from Figure~\ref{phylogeny}. We find neat coelacanth stuff!}
\end{figure}



\end{document}