\documentclass[fleqn,10pt]{wlpeerj}
\title{Comprehensive, structurally-curated alignment and phylogeny of vertebrate biogenic amine receptors}

\author[1,2,3]{Stephanie J. Spielman}
\author[1,2,3]{Keerthana Kumar}
\author[1,2,3]{Claus O. Wilke}
\affil[1]{Department of Integrative Biology, The University of Texas at Austin, Austin, U.S.A.}
\affil[2]{Institute of Cellular and Molecular Biology, The University of Texas at Austin, Austin, U.S.A.}
\affil[3]{Center for Computational Biology and Bioinformatics, The University of Texas at Austin, Austin, U.S.A.}




\keywords{biogenic amine receptors, G-protein coupled receptors, multiple sequence alignment, phylogenetics, protein evolution}


\begin{abstract}
Biogenic amine receptors play critical roles in regulating behavior and physiology, particularly within the central nervous system, in both vertebrates and invertebrates. These receptors belong to the G-protein coupled receptor (GPCR) family and interact with endogenous bioamine ligands, such as dopamine, serotonin, and epinephrine, and they are targeted by a wide array of pharmaceuticals. Despite these receptors' clear clinical and biological importance, their evolutionary history remains poorly characterized. In particular, the relationships among biogenic amine receptors and any specific evolutionary constraints acting within distinct receptor subtypes are largely unknown. To advance and facilitate studies in this receptor family, we have constructed a comprehensive, high-quality sequence alignment of vertebrate biogenic amine receptors. In particular, we took \textbf{measures} to ensure that alignment columns capture the highly-conserved GPCR structural domains, and we demonstrate how ignoring structural information can produce spurious homology inferences. Using this alignment, we further constructed a structurally-partitioned maximum-likelihood phylogeny, from which we deduce novel biogenic amine receptor relationships and uncover previously unrecognized lineage-specific receptor clades. Moreover, we find that roughly 1\% of the 3039 sequences in our final alignment are either misannotated or unclassified, and we propose updated classifications for these receptors. We release our comprehensive alignment and its corresponding phylogeny as a resource for future research into the evolution and diversification of biogenic amine receptors.
\end{abstract}

\begin{document}

\flushbottom
\maketitle
\thispagestyle{empty}


\section*{Introduction}

Biogenic amines, such as the molecules serotonin and dopamine, play critical roles in virtually all Metazoans and exert significant influence on both behavior and physiology. In vertebrates, the biogenic amine receptor family, which includes dopamine (DRD), histamine (HRH), trace (TAAR), adrenergic (ADR), muscarinic cholinergic (mAChR), and most serotonin (5HTR) receptors, primarily mediates biogenic amine activity.  Biogenic amine receptors belong to the broad family of G~protein-coupled receptors (GPCRs), one of the largest and most diverse eukaryotic receptor families. Indeed, due to the extensive diversity of biological functions they direct and the ongoing expansion of their ligand repertoire, GPCRs are considered one of the most evolutionarily innovative and successful gene families \citep{BockaertPin1999,Lagerstrom2008}.

Biogenic amine receptors form a clade within the large Rhodopsin-like GPCR family \citep{Fredrikssonetal2003,KakaralaJamil2014}, whose emergence likely accompanied that of the Opisthokont (Fungi and Metazoa) lineage \citep{Krishnan2012}. The Rhodopsin-like family expanded substantially in Metazoa, and the specific diversification of biogenic amine receptors has contributed significantly to central nervous system functioning \citep{Callieretal2003,Nichols2008}. Like all GPCRs, biogenic amine receptors have a charcterisitic, highly-conserved structure of seven transmembrane (TM) domains separated by three extracellular (ECL) and three intracellular (ICL) loops, and they propagate intracellular signaling through a G~protein-mediated pathway. Moreover, these receptors are prominent targets for a wide range pharmaceuticals aimed to treat myriad diseases such as schizophrenia, migraines, hypertension, allergies and asthma, and stomach ulcers \citep{Schoneberg2004,Eversetal2005,Masonetal2012}.

In spite of these receptors' biological and clinical importance, studies on their evolution are limited and have predominantly focused on individual receptor subtypes, namely TAAR \citep{Gloriametal2005,Lindemann2005,Hashiguchi2007}, DRD \citep{Callieretal2003,Yamamotoetal2013}, and 5HTR \citep{Anbazhagan2010}. Moreover, many of these studies, and indeed studies on the general evolution of the Rhodopsin-like family, have examined very narrow species distributions, for instance specifically teleosts \citep{Gloriametal2005}, primates \citep{Anbazhagan2010}, humans and mice \citep{Vassilatis2003,KakaralaJamil2014}, or even strictly humans \citep{Fredrikssonetal2003}. Thus, virtually no studies accounting for the full breadth of vertebrate bioamine receptor sequences have been conducted.

To gain a comprehensive understanding of this receptor family's evolution, a high-quality multiple sequence alignment (MSA) is needed. MSAs provide the foundation for nearly all comparative sequence analyses, and they are commonly used to locate conserved sequence motifs, identify functionally important residues, and investigate evolutionary histories. As constructing an MSA represents the first step in any sequence analysis, MSA errors are known to bias these downstream analyses \citep{Ogden2006, Wong2008, Jordan2012}. It is therefore crucial to ensure accuracy in MSAs to the extent possible. 

For GPCR sequences, in particular, any MSA should recapitulate the canonical seven-TM structure, which a naive alignment of sequences cannot necessarily accomplish. Several varieties of MSA software platforms have been developed that incorporate structural information into the alignment algorithm by aligning sequences to a given protein crystal structure \citep{promals3d, 3dcoffee} or hidden Markov model (HMM) profile \citep{hmmer, Chang2012, Hill2012}. In fact, some programs, such as MP-T \citep{Hill2012} and TM-Coffee \citep{Chang2012} cater specifically to membrane proteins. However, all of these programs are extremely computationally-intensive and ill-suited for large-scale applications, and they incur prohibitive runtimes with large datasets. Furthermore, many of these programs require the use of a single crystal structure or HMM profile to guide sequence alignment. While all GPCRs contain seven TM domains, different GPCR subfamilies, particularly the biogenic amine receptors, feature a wide variety of ICL and ECL sizes. For example, human HRH1 and DRD3 contain roughly 27 and 117 residues, respectively, in their ECL3 domains, and roughly 68 and 14 residues, respectively, in their ICL3 domains [as predicted by GPCRHMM \citep{Wistrand2006}]. Thus, aligning diverse sequences using a single structure may not effectively capture the domain variability across biogenic amine receptor subtypes. 

Here, we integrate a traditional progressive alignment approach with structural predictions to generate a high-quality, comprehensive (3039 sequences) MSA of vertebrate biogenic amine receptors, representing the most extensive such dataset to date. Our alignment approach ensures that the GPCR domains aligned correctly across all sequences. We used this MSA to construct a maximum likelihood (ML) phylogeny of vertebrate biogenic amine receptors, and we found that a paritioned phylogeny which separately considered transmembrane and extramembrane domains strongly improves upon an unpartitioned phylogeny. Using this structurally-partitioned phylogeny, we were able to discern relationships among biogenic amine receptor subtypes with a far increased level of sensitivity relative to previous studies, as well as identify novel lineage-specific receptor clades and clarify NCBI annotations for over 30 sequences.

We present this vertebrate biogenic amine receptor MSA and its corresponding phylogeny as a resource for any group interested in studying the dynamic evolutionary processes, structural, and/or functional constraints operating within this class of GPCRs. All data, including MSAs, phylogenies, and sequence descriptions, as well as all code used to generate these data, are freely available from \\\texttt{https://github.com/sjspielman/amine\_receptors}. We expect that these data will prove extremely useful for studying both the broad patterns governing biogenic amine receptor sequence evolution and the evolutionary trends specific to certain receptor subtypes. Further, our curated MSA should serve as a helpful resource in the ongoing development of homology models and pharmaceutical therapeutics targeting these receptors \citep{Kristiansen2004,Ishiguro2004,Eversetal2005,Masonetal2012}.




\section*{Results and Discussion}

\subsection*{Constructing a structurally-curated MSA of biogenic amine receptors}

We collected all sequences using PSI-BLAST from the RefSeq database \citep{refseq} and removed all poor-quality sequences as well as  sequences with excessive ambiguities (see \emph{Methods} for details). We then filtered this dataset to remove any sequences which could not be robustly classified as GPCRs, leaving a total dataset of 3464 sequences. In particular, we used the software GPCRHMM \citep{Wistrand2006}, which uses a hidden markov model approach to identify GPCRs from protein sequence data alone, to ascertain if each sequence was indeed a GPCR. GPCRHMM features an exceptionally low false positive rate ($~\approx 1\%$) as well as a 15\% increase in sensitivity relative to other similar structural prediction programs \citep{Wistrand2006}. 

In addition to identifying GPCR sequences, GPCRHMM can also predict transmembrane domain regions for Rhodopsin-like GPCRs with exceptional accuracy \citep{SpielmanWilke2013}. Indeed, as shown in Figure~\ref{pdb_gpcrhmm}, GPCRHMM yields excellent domain predictions for resolved amine receptor crystal structures, and thus GPCRHMM serves as a robust proxy for more computationally-intensive structural predictors. Therefore, we used GPCRHMM to assign every residue in all 3464 protein sequences to its respective structural domain (extracellular, transmembrane, or intracellular). 

We then aligned these protein sequences with MAFFT \citep{mafftv7} and we assessed how well the resulting MSA recapitulated the overarching GPCR domain structure using the GPCRHMM domains predictions. To this end, we determined the consensus structural domain for each MSA column and assessed how well this naive MSA captured structural information (Figure~\ref{domains}A). We found that, although we had already filtered out putatively non-GPCR sequences, several hundred sequences did not align according to the overarching structure. Many sequences' domains were shifted out of structural frame, causing TM domains to inappropriately align with loop domains, or vice versa, ultimately preventing MSA columns from being truly homologous. Moreover, the mere presence of these misaligned sequences in the naive MSA introduced a substantial amount of gaps, often times within a single TM domain (notably TM1 and TM7, as seen in Figure~\ref{domains}A). While gaps are not inherently problematic in MSAs, the highly-conserved GPCR structure means that there should not be insertion/deletion events within transmembrane regions. In particular, these confounding sequences led oftentimes disrupted the GPCR structure, as highlighted by gaps introduced in the middle of a TM domain.

We therefore adopted an iterative strategy to systematically cull poorly-aligned sequences. As outlined outlined in Figure~\ref{flowchart} we aligned protein sequences with MAFFT \citep{mafftv7}. Using the residue domain assignments computed with GPCRHMM, we determined the consensus domain for each column in this MSA. Next, we discarded all sequences for which $\geq 5\%$ of residues did not correspond to their respective column's consensus domain. We realigned the remaining sequences with MAFFT and continued in this manner until no more sequences were discarded. Importantly, this strategy did not require any manual data filtering or visual data inspection, thus avoiding any confounding subjectivity in MSA processing. The final structurally-curated MSA contained a total of 3039 sequences, which were broadly distributed across receptor subtypes (Table~\ref{tab:abbrev_count}) and vertebrate taxa (Figure~\ref{taxa_dist}). 

From this final MSA, we additionally created a masked MSA, in which protein residues which did not conform to their respective consensus domains were replaced with a ``?''. By masking these positions with an ambiguous character, we ensured as much as possible that each MSA column only contained residues belonging to the same structural domain. In total, 2.69\% of all MSA positions were masked for this MSA.


\subsection*{Structurally-aware MSA strongly improves phylogenetic inference}

We used the masked MSA to construct ML phylogenies in RAxML. 

To demonstrate the utility of analyzing biogenic amine receptors with a structurally-curated MSA, we constructed five distinct maximum likelihood (ML) phylogenies in RAxML. We built a phylogeny using the naive MSA, and we built two phylogenies each from the structural unmasked and masked MSAs. Previous work has shown that combined structural and functional constraints impose differing selection pressures in TM vs.\ extramembrane (EM) domains, in turn producing distinct amino-acid frequency distributions in each domain class \citep{Tourasse2000,Stevens2001,Julenius2006,Oberai2009,SpielmanWilke2013,FranzosaXueXia2013}. As our structurally-curated MSAs allowed us to precisely identify each MSA column as either TM or EM, we were able to conduct far more rigorous phylogentic inference using a partitioned analysis. Therefore, for each of our two structurally-curated MSAs (masked and unmasked), we inferred two ML phylogenies: one with two partitions representing TM and EM columns and one with a single partition for the entire MSA. 

As assessed by AIC scores, the structurally-curated masked MSA yielded a far superior phylogeny compared to all other MSAs (Table~\ref{tab:phylo_AIC}), highlighting the benefits of analyzing GPCRs in a structurally-aware context. This finding also underscores that any structurally-aware study must be undertaken cautiously. While partitioning the MSA based on structural domains was clearly beneficial, ensuring that each MSA column strictly contained residues belonging to the same domain was critical. Having even a few TM residues in a column assigned to the EM partition, or vice versa, strongly hindered phylogenetic fit.

That the masked structural MSA produced a phylogeny with far better fit than did the unmasked structural MSA additionally reveals the potential benefits to MSA filtering, which entails removing putatively unreliable positions and/or columns from MSAs. In spite of a large body of literature investigating the utility of MSA filtering, the particular circumstances under which filtering is beneficial remain ambiguous \citep{Castresana2000,Talavera2007,Schloss2010,Penn2010,Jordan2012,Privman2012,Wu2012,SpielmanDawsonWilke2014}. Here, we find that masking residues based on biological information regarding protein structure can greatly improve phylogenetic fit (Table~\ref{tab:phylo_AIC}), indicating that residue filtering, at least in this circumstance, successfully increased phylogenetic signal. However, we caution that standard MSA filtering algorithms may not always lead to such improvements. For instance, the underlying problem with our structurally-naive MSA was likely not misaligned residues or columns, but rather the presence of confounding sequences which would not align properly with the overarching GPCR structure. Masking particular MSA regions would not have addressed this issue, as confounding sequences would have remained in the MSA even after filters were applied. Therefore, while MSA filtering may be beneficial, residue masking alone cannot fix a poorly-constructed MSA. 



\subsection*{Structurally-aware phylogeny reveals unknown biogenic amine receptor relationships and clades}

Our resulting phylogeny, shown in Figure~\ref{phylogeny}, represents the most comprehensive and curated vertebrate biogenic amine receptor phylogeny to date. This tree broadly captures many known features of biogenic amine receptor evolution, in particular that these receptors do not cluster based on ligand-binding but rather have undergone extensive functional convergent evolution. Indeed, our phylogeny reveals that only two ligand-based receptor classes, mAChR and TAAR, are truly monophyletic. 

Our phylogeny features remarkably high bootstrap support for each distinct clade of receptor subtypes. We additionally find very strong support for three deeper nodes in the phylogeny that reveal the relationships among distinct receptor subtypes. The first contains the three clades HRH1, mAChR, and HRH-3,4, the second contains the clades 5HTR-1, 5HTR-5, and 5HTR-7, and the third contains the 5HTR-4 and TAAR clades. Previous studies have yielded conflicting phylogenetic placements for the 5HTR-7 clade; some have argued that 5HTR-7 is phylogenetically distinct from all other 5HTR sequences \citep{KakaralaJamil2014}, while others have found evidence for a single clade containing 5HTR-5,7 as a sister taxa to a clade containing ADRA1 sequences \citep{Fredrikssonetal2003}. Alternatively, we find moderate-to-strong support for the 5HTR-7 clade having originated before subsequent diversification into 5HTR-5 and 5HTR-1, and we find full support showing that ADRA1 forms an entirely distinct monophyletic group outside all other vertebrate biogenic amine receptors. In addition, as previously mentioned, our phylogeny reveals that HRH-3,4 is actually a single monophyletic group. While the HRH-4 clade contains strictly mammalian sequences, including monotreme (platypus) sequences, HRH-3 sequences are broadly distributed across vertebrate taxa. We therefore hypothesize that HRH-4 arose from an HRH-3 duplication concurrent with mammalian origins.

In addition, among the 3039 sequences in structurally-curated MSA, we identified 31 sequences (~1\%) that we considered misannotated (Table~\ref{tab:classif}), either because the NCBI annotation did not match the sequences' phylogenetic placement or the sequences did not cluster with known biogenic amine receptor types. Several NCBI annotations identified the correct receptor class but the incorrect receptor subtype, whereas other sequences were entirely uncharacterized. We additionally uncovered an entirely unknown clade of biogenic amine receptors. This unknown clade, sister to HRH2, only contains avian sequences and a single \emph{Xenopus tropicalis} (western-clawed frog) sequence. Thus, it is likely that this clade emerged concurrently with tetrapods and was secondarily lost in reptiles/birds and mammals. Interestingly, all but one of this clade's sequences were annotated in NCBI as either octopamine or No9-like receptors, both of which are insect-specific biogenic amine receptors that do not occur in vertebrate taxa \citep{Roeder2005}. The last sequence, alternatively, was annotated as 5HTR-7-like. Taken together, these sequence misannotations suggest an intriguing hypothesis that this clade evolved from an ancestral 5HTR sequence, and subsequent convergent evolution to insect-specific biogenic amine receptors has allowed these receptors to interact with atypical ligands for vertebrates.


\subsection*{Dynamic lineage-specific evolution of the trace-amine associated receptors}
Of particular interest in our phylogeny are the unique evolutionary patterns revealed within the TAAR clade. While all TAAR sequences do cluster together, the TAAR phylogeny reveals the extensive expansion and contraction events characterizing this receptor family's evolution \citep{Lindemann2005,Hashiguchi2007,Staubert2010,Staubert2013}. In fact, the TAAR subtree, displayed in Figure~\ref{taar_tree}, differs somewhat from previously proposed TAAR phylogenies \citep{Lindemann2005, Hashiguchi2007}. Figure~\ref{taar_tree} highlights the dynamic species distributions across TAAR subtypes. In particular, the presence of several lineage-specific subclades as well as unresolved subclades generate novel hypotheses regarding TAAR subtype origins. 

1. the clade containing TAAR -6, -7, -8, and -9 appears to have undergone repeated expansion and contraction events, with certain clades present only in meta+euth mammals and others present only in crocs+lizards. [rampant diversification within this subclade]
2. mammals have a lot of lineage-specific diversity. for instance, there is a caprid-specific and rodent-specific clade, consistent with evolution of chemosensory receptors
3. amphibian seqeunces are notably absent from this phylogeny. for example, several cases where lobe-finned fish sequences are outgroup to reptile/bird+mammal clades, or indeed where they are all in a single well-supported clade but amphibian sequences aremissing. taars are likely involved in chemosensory processes, so it is possible that amphibia have invested this process in other receptors rather than taars and thus have lost these sequences.
4. taar-12 + taar-4 cluster together. probably the ray-finned are super different, but importantly this is only only subclade of taars we could identify as present across all euteleostii.
5. Furthermore, several lobe-finned fish (coelacanth) sequences are scattered across the TAAR tree and do not clearly cluster with any TAAR subtypes, likely reflecting this lineage's ancient divergence and unique evolutionary trajectory \citep{coelacanth2013}.



 The phylogenetic distribution of lobe-finned fish sequences may aid futur e endeavors to tease apart evolutionary origins of certain TAAR subtypes, specifically whether they represent teleost-specific duplications \citep{Gloriametal2005} or whether they represent ancient TAARs that emerged before teleost divergence but were secondarily lost in lobe-finned fish and/or tetrapods.

In addition, a small clade sister to TAAR (labeled in Figures \ref{phylogeny} and \ref{taar_tree} as TAAR$^\ast$) strictly contains sequences annotated by NCBI as ``5HTR-4-like.'' At first glance, these annotations might suggest that 5HTR-4 is in fact paraphyletic, diversifying gradually before giving rise to TAARs. However, as all sequences in TAAR$^\ast$ belong taxonomically either to teleost or \emph{Xenopus tropicalis}, we suspect that this clade actually corresponds to the so-called TAAR-V cluster identified by \cite{Hashiguchi2007}. Indeed, the TAAR-V cluster contains a similar taxonomic distribution to our TAAR$^\ast$ and constitutes an outgroup to all other vertebrate TAAR sequences, as our phylogeny similarly displays.

\subsection*{Phylogenetic methods alone do not suffice to infer the evolutionary history of biogenic amine receptors}
Although we were able to identify several new features of biogenic amine receptor evolution, the majority of deeper splits in the phylogeny had very low bootstrap support, meaning that most of the broader relationships among biogenic amine receptors remain unresolved. This result highlights that a strictly phylogenetic approach cannot fully elucidate the complex evolutionary histories of expanding gene families. In particular, modern phylogenetic methods focus solely on the substutition process and treat MSA gaps simply as missing data. However, gaps actually represent the evolutionary events of insertion and deletions (indels), and some have suggested that ignoring this information ultimately hinders phylogenetic accuracy \citep{Morrison2008,Loytynoja2008,Warnow2012,Luanetal2013}. 

This limitation is especially problematic for GPCRs. Following duplication events, GPCRs appear to experience major indel events in their ICL and/or ECL domains, leading to dramatic shifts in loop domain sizes during the sub/neofunctionalization process. Unfortunately, the evolutionary intermediates that existed during these domain transitions have long-since disappeared from genomes, and there is no obvious way to infer the sequences of these missing links. Although the substitution process is key for understanding GPCR evolution, fully classifiying relationships among GPCR families requires some understanding of how these radical domain changes occur. Therefore, additional approaches, such as syntenic analyses \citep{Sundstrom2010,Widmark2011,YegorovGood2012,Hwangetal2013}, combined with the phylogeny presented here should prove useful towards resolving the complete evolutionary history of vertebrate biogenic amine receptors. 


\section*{Conclusions}

We have established a comprehensive, high-quality, structurally-curated MSA of vertebrate biogenic amine receptors. We hope that this MSA, along with its ML phylogeny, will serve as a robust resource for future studies investigating the evolutionary dynamics as well as structural/functional constraints operating within distinct receptor clades or indeed universal patterns that generally govern biogenic amine receptor evolution. Future work may seek to combine the analyses we have accomplished here with syntenic or molecular clock approaches to elucidate receptors' origin and precise evolutionary trajectories. Moreover, our curated MSA should prove useful in increasing accuracy in homology modeling and/or pharmaceutical development for these clincally important receptors \citep{Kristiansen2004,Ishiguro2004,Eversetal2005,Masonetal2012}.




\section*{Methods}

\subsection*{Sequence Collection and Processing}
We collected protein sequences using PSI-BLAST \citep{psiblast}, specifically from the RefSeq (v2.2.29+) database \citep{refseq}, for 42 distinct human biogenic amine receptor sequences representing the full range of known receptors in the human genome. To obtain distant yet well-supported orthologs, we ran each PSI-BLAST search for 5 iterations with an e-value cutoff of $10^{-20}$, a sequence identity threshold of 25\%, and a length difference of $\pm50$\% relative to the seed sequence. After combining all sequences recovered from the individual PSI-BLAST searches, we discarded duplicate sequences, leaving a total of 4232 PSI-BLAST results. We then filtered this sequence set to remove sequences from non-vertebrate taxa, sequences annotated as low-quality, pseudogene, and/or partial, and sequences which contained more than 1\% ambiguous residues (i.e.\ B, X, or Z). We additionally used the program GPCRHMM \citep{Wistrand2006} to determine whether a given sequence was indeed a GPCR. We discarded sequences which had either a local or global GPCRHMM score less than 10, both extremely conservative thresholds. Thus, while it is possible that some true GPCRs were discarded, these stringent thresholds for both local and global scores provide high confidence that all retained sequences were indeed GPCRs. Together, these filters left a total of 3464 receptor sequences.


\subsection*{Sequence Alignment and Phylogenetic Reconstruction}
Before aligning sequences, we used the program GPCRHMM \citep{Wistrand2006} to assign each residue in all protein sequences to its respective structural domain (extracellular, transmembrane, or intracellular) using a $0.5$ posterior probability cutoff. We then aligned and filtered sequences according to the strategy outlined in Figure~\ref{flowchart}, which specifically employed MAFFT v7.149b \citep{mafftv7}. 

All phylogenies were created using RAxML v8.1.1 \citep{raxml} using the LG+F \citep{LG} amino acid exchangability matrix with empirical amino acid frequencies and the CAT model of site heterogeneity \citep{Stamatakis2006}, with the default 25 rate categories. For inferences incorporating structural partitions, we assigned each partition a unique evolutionary model using these settings. Thus, due to the +F parameterization, the different partitions were allowed to have distinct equilibrium frequencies. Final parameter values for all phylogenetic inferences were optimized with the GAMMA model of heterogeneity. We performed 200 bootstrap replicates for each phylogeny.



\section*{Acknowledgments}
This work was supported in part by NIH grant R01 GM088344, ARO grant W911NF-12-1-0390, DTRA grant HDTRA1-12-C-0007, and NSF Cooperative Agreement No. DBI-0939454 (BEACON Center).  Computational resources were provided by the University of Texas at Austin's Center for Computational Biology and Bioinformatics (CCBB). We would like to thank Ahmad R. Sedaghat, MD, PhD for suggesting biogenic amine receptor evolution as a worthwhile study system.


\bibliography{bibliography}


\newpage


\section*{Figures and Tables}


\vspace*{5cm}


\begin{table}[htbp]
	\centering
	\begin{tabular}{l l l}
		\hline\noalign{\smallskip}
		\multicolumn{1}{c}{Receptor Class} & \multicolumn{1}{c}{Abbreviation} & \multicolumn{1}{c}{N} \\
		\hline\noalign{\smallskip}
		Serotonin receptors & \quad 5HTR & 972  \\
		Trace amine-associated receptors & \quad TAAR & 343 \\
		Histamine receptors & \quad HRH & 286 \\
		Muscarinic cholinergic receptors & \quad mAChR & 353  \\
		Adrenergic receptors & \quad ADR & 611  \\
		Dopamine receptors & \quad DRD & 464 \\
		Unknown receptors & \quad Unknown & 10 \\
		\noalign{\smallskip}\hline\noalign{\smallskip} 
	\end{tabular}
	\caption{\label{tab:abbrev_count} Biogenic amine receptor classes, and their abbreviations, considered in this study. The receptor class ``Unknown'' refers to the corresponding uncharacterized clade in Figure~\ref{phylogeny}, and the column ``N'' indicates the total number of sequences for each broad receptor class in our structutally-curated MSA.}
\end{table}



\vspace*{3cm}

\begin{table}[htbp]
	\centering
	\begin{tabular}{l c l l c}
		\hline\noalign{\smallskip}
		\multicolumn{1}{c}{MSA} & \multicolumn{1}{c}{Partitioned} & \multicolumn{1}{c}{$\ln L$} & \multicolumn{1}{c}{k} & \multicolumn{1}{l}{$\Delta$AIC} \\
		\hline\noalign{\smallskip}
		Structural Masked & Yes & -505500.8 & 6115 & 0 \\
		Structural Masked & No & -515991.7 & 6095 & 1752 \\  
		Structural Unmasked & Yes & -515343.6 & 6115 & 19685 \\
		Structural Unmasked & No & -515991.7 & 6095 & 20941 \\ 
		Naive & No &  -589703.7 & 6945 & 170047 \\
		\noalign{\smallskip}\hline\noalign{\smallskip} 
	\end{tabular}
	\caption{\label{tab:phylo_AIC} $\Delta$AIC scores relative to the best performing for phylogenies. The column labeled ``Partitioned'' indicates whether phylogenetic inference was conducted with distinct TM (transmembrane) and EM (extramembrane) partitions. AIC is computed as AIC $= 2(k - \ln L)$, where $k$ is the number of free parameters of the model, and $\ln L$ is the log-likelihood \citep{Akaike1974,BurnhamAnderson2004}. AIC scores are reported here relative to the phylogeny with the lowest AIC score (structural masked with partitions), representing the best-fitting phylogeny.}
\end{table}


\newpage

\begin{table}[htbp]
	\centering
	\begin{tabular}{l l l l l l}
		\hline\noalign{\smallskip}
		\multicolumn{1}{c}{Protein ID} & \multicolumn{1}{c}{Nucleotide ID} & \multicolumn{2}{c}{Current Classification} & \multicolumn{2}{c}{Proposed Classification} \\
		\hline\noalign{\smallskip}
		XP\_005797918.1 & XM\_005797861.1 & \qquad \qquad & DRD-2 & \qquad \qquad & DRD-3 \\
		XP\_003967971.1 & XM\_003967922.1 & \qquad \qquad & DRD-2 & \qquad \qquad & DRD-3 \\
		NP\_001266433.1 & NM\_001279504.1 & \qquad \qquad & mAChR-4 & \qquad \qquad & mAChR-2 \\
		XP\_001520508.2 & XM\_001520458.3 & \qquad \qquad & HRH-3 & \qquad \qquad & HRH-4 \\
		XP\_005282846.1 & XM\_005282789.1 & \qquad \qquad & HRH-4 & \qquad \qquad & HRH-3 \\
		XP\_001920844.1 & XM\_001920809.1 & \qquad \qquad & TAAR-4-like & \qquad \qquad & TAAR-12 \\
		NP\_001076571.1 & NM\_001083102.1 & \qquad \qquad & TAAR-64 & \qquad \qquad & TAAR-13 \\
		XP\_006014096.1 & XM\_006014034.1 & \qquad \qquad & TAAR-9-like & \qquad \qquad & TAAR-4 \\
		XP\_003201718.2 & XM\_003201670.2 & \qquad \qquad & TAAR-1-like & \qquad \qquad & TAAR-10 \\
		NP\_001076546.1 & NM\_001083077.1 & \qquad \qquad & TAAR-11-like & \qquad \qquad & TAAR-10 \\
		NP\_001083418.1 & NM\_001089949.1 & \qquad \qquad & uncharacterized & \qquad \qquad & ADRB \\
		NP\_001103208.1 & NM\_001109738.1 & \qquad \qquad & uncharacterized & \qquad \qquad & HRH-2 \\
		NP\_001124143.1 & NM\_001130671.1 & \qquad \qquad & uncharacterized & \qquad \qquad & TAAR-12 \\
		XP\_001337671.1 & XM\_001337635.2 & \qquad \qquad & 5HTR-4-like & \qquad \qquad & TAAR$^\ast$ \\
		XP\_003976403.1 & XM\_003976354.1 & \qquad \qquad & 5HTR-4-like & \qquad \qquad & TAAR$^\ast$ \\
		XP\_005810466.1 & XM\_005810409.1 & \qquad \qquad & 5HTR-4-like & \qquad \qquad & TAAR$^\ast$ \\
		XP\_003454279.1 & XM\_003454231.1 & \qquad \qquad & 5HTR-4-like & \qquad \qquad & TAAR$^\ast$ \\
		XP\_004549625.1 & XM\_004549568.1 & \qquad \qquad & 5HTR-4-like & \qquad \qquad & TAAR$^\ast$ \\
		XP\_002935532.2 & XM\_002935486.2 & \qquad \qquad & 5HTR-4-like & \qquad \qquad & TAAR$^\ast$ \\
		XP\_006013317.1 & XM\_006013255.1 & \qquad \qquad & 5HTR-4-like & \qquad \qquad & TAAR$^\ast$ \\	
		XP\_005510029.1 & XM\_005509972.1 & \qquad \qquad & 5HTR-7-like & \qquad \qquad & Unknown \\
		XP\_002187301.2 & XM\_002187265.2 & \qquad \qquad & Octopamine receptor-like & \qquad \qquad & Unknown \\
		XP\_002937327.2 & XM\_002937281.2 & \qquad \qquad & Octopamine receptor-like & \qquad \qquad & Unknown \\
		XP\_005045681.1 & XM\_005045624.1 & \qquad \qquad & Octopamine receptor-like & \qquad \qquad & Unknown \\
		XP\_005144673.1 & XM\_005144616.1 & \qquad \qquad & Octopamine receptor-like & \qquad \qquad & Unknown \\
		XP\_005229932.1 & XM\_005229875.1 & \qquad \qquad & Octopamine receptor-like & \qquad \qquad & Unknown \\
		XP\_005428400.1 & XM\_005428343.1 & \qquad \qquad & Probable GPCR No9-like & \qquad \qquad & Unknown \\
		XP\_005490920.1 & XM\_005490863.1 & \qquad \qquad & Probable GPCR No9-like & \qquad \qquad & Unknown \\
		XP\_005518128.1 & XM\_005518071.1 & \qquad \qquad & Probable GPCR No9-like & \qquad \qquad & Unknown \\
		XP\_006111669.1 & XM\_006111607.1 & \qquad \qquad & Octopamine receptor-like & \qquad \qquad & Unknown \\
		XP\_420867.2 & XM\_420867.4 & \qquad \qquad & Octopamine receptor & \qquad \qquad & Unknown \\
		\noalign{\smallskip}\hline\noalign{\smallskip} 
	\end{tabular}
	\caption{\label{tab:classif} Misannotated and uncharacterized sequences identified through phylogenetic analysis. Based on sequence placement in the structurally-curated phylogeny (Figure~\ref{phylogeny}) we propose updated classifications for 31 biogenic amine receptor sequences. The proposed classifications ``Unknown'' and ``TAAR$^\ast$'' refer to the corresponding clades in Figure~\ref{phylogeny}.}
\end{table}

\newpage

\begin{figure}[htbp]
	\centerline{\includegraphics[width=15cm]{figures/pdb_gpcrhmm.pdf}}
	\caption{\label{pdb_gpcrhmm} GPCRHMM domain predictions for representative biogenic amine receptor crystal structures from the Protein Data Bank (PBD). Gene names are shown in capital letters above each structure, and corresponding PDB IDs are shown in parentheses. Dark blue represents predicted extracellular residues, red represents predicted TM residues, light blue represents predicted intracellular residues.}
\end{figure}



\begin{figure}[htbp]
	\centerline{\includegraphics[width=18cm]{figures/alignment_flowchart.pdf}}
	\caption{\label{flowchart} Iterative alignment strategy to create a structurally-curated MSA of vertebrate biogenic amine receptors. A total of 3464 sequences were initially input (``Validated GPCR protein sequences''), and the final MSA contained 3039 protein sequences. Residues marked with ``I'' represent intracellular residues, those marked with ``M'' represent transmembrane residues, and those marked with ``E'' represent extracellular residues. MSA gaps were treated as missing data when determining each column's consensus structural domain. Sequenced were removed (``remove highly discordant sequences'') if $\geq 5\%$ of columns belonged to a different structural domain than the respective consensus domain. Note that the MSA shown in this figure represents a subset of the entire MSA.}
\end{figure}



\begin{figure}[htbp]
	\centerline{\includegraphics[width=7cm]{figures/taxonomic_distribution.pdf}}
	\caption{\label{taxa_dist} Cladogram of the taxonomic distribution of all sequences in the final structurally-curated MSA. All sequences belonged to the Euteleostomi clade of jawed vertebrates. Numbers in parentheses indicate the total number of sequences from the respective clade. We note that our MSA is particularly enriched for sequences from Eutherian (placental mammal) species, likely due to the stringent filters we applied to sequence collection that favored fully-sequenced genomes.}
\end{figure}

\vspace*{4cm}


\begin{figure}[htbp]
	\centerline{\includegraphics[width=7in]{figures/domains_struc_naive.png}}
	\caption{\label{domains} Graphical representation of a subset of the naive and curated biogenic amine receptor MSAs. Each image displays 130 MSA rows focused specifically on the MSA section containing the seven TM domains. Dark blue represents predicted extracellular residues, red represents predicted TM residues, lighter blue represents predicted intracellular residues, and gray represents MSA gaps. The bottom bar below each MSA figure shows the consensus domain structure for each MSA. Note that all columns which contain only gaps in this subset of sequences have been removed from this figure for visual clarity.}
\end{figure}

\newpage

\begin{figure}[htbp]
	\centerline{\includegraphics[width=18cm]{figures/vert_amine_tree.pdf}}
	\caption{\label{phylogeny} Maximum likelihood phylogeny of vertebrate biogenic amine receptors built using the masked structural MSA in RAxML. Nodes with open circles indicate $\geq 50\%$ bootstrap support, and nodes with closed circles indicate $\geq 90\%$ bootstrap support. Biogenic amine receptors are abbreviated as in Table~\ref{tab:abbrev_count}. The clade labeled ``Unknown'' could not be clearly identified as one of the major receptor types and may represent a previously unrecognized biogenic amine receptor clade.}
\end{figure}


\newpage

\begin{figure}[htbp]
	\centerline{\includegraphics[width=15cm]{figures/taar_revision_taxonomy.pdf}}
	\caption{\label{taar_tree} \textbf{REDO LEGEND!!!}Subclade of the TAAR receptors within the phylogeny shown in Figure~\ref{phylogeny}. Nodes with open circles indicate $\geq 50\%$ bootstrap support, and nodes with closed circles indicate $\geq 90\%$ bootstrap support. \color{blue}{include caprid, rodent specific clade information somehow!} }
\end{figure}


\end{document}
