\documentclass[fleqn,10pt]{wlpeerj}
%DIF LATEXDIFF DIFFERENCE FILE
%DIF DEL original_amine_paper.tex   Tue Dec 23 16:14:57 2014
%DIF ADD amine_paper.tex            Tue Dec 23 16:51:18 2014
%DIF 2c2-3
%DIF < \title{Comprehensive, structurally-curated alignment and phylogeny of vertebrate biogenic amine receptors}
%DIF -------
\usepackage{hyperref} %DIF > 
\title{Comprehensive, structurally-informed alignment and phylogeny of vertebrate biogenic amine receptors} %DIF > 
%DIF -------

\author[1,2,3]{Stephanie J. Spielman}
\author[1,2,3]{Keerthana Kumar}
\author[1,2,3]{Claus O. Wilke}
\affil[1]{Department of Integrative Biology, The University of Texas at Austin, Austin, U.S.A.}
\affil[2]{Institute of Cellular and Molecular Biology, The University of Texas at Austin, Austin, U.S.A.}
\affil[3]{Center for Computational Biology and Bioinformatics, The University of Texas at Austin, Austin, U.S.A.}




%DIF 14c15
%DIF < \keywords{biogenic amine receptors, G-protein coupled receptors, multiple sequence alignment, phylogenetics, protein evolution}
%DIF -------
\keywords{biogenic amine receptors, G-protein coupled receptors, multiple sequence alignment, phylogenetics} %DIF > 
%DIF -------


\begin{abstract}
%DIF 18c19
%DIF < Biogenic amine receptors play critical roles in regulating behavior and physiology, particularly within the central nervous system, in both vertebrates and invertebrates. These receptors belong to the G-protein coupled receptor (GPCR) family and interact with endogenous bioamine ligands, such as dopamine, serotonin, and epinephrine, and they are targeted by a wide array of pharmaceuticals. Despite these receptors' clear clinical and biological importance, their evolutionary history remains poorly characterized. In particular, the relationships among biogenic amine receptors and any specific evolutionary constraints acting within distinct receptor subtypes are largely unknown. To advance and facilitate studies in this receptor family, we have constructed a comprehensive, high-quality, structurally-curated sequence alignment of vertebrate biogenic amine receptors. We demonstrate that aligning GPCR sequences without considering structure produces an alignment with substantial error, whereas a structurally-aware approach greatly improves alignment accuracy. Moreover, we show that phylogenetic inference with our structurally-curated alignment offers dramatic improvements over a structurally-naive alignment. Using the structural alignment and its corresponding phylogeny, we deduce novel biogenic amine receptor relationships and uncover previously unrecognized lineage-specific receptor clades. Moreover, we find that roughly 1\% of the 3039 sequences in our final alignment are either misannotated or unclassified, and we propose updated classifications for these receptors. We release our comprehensive alignment and its corresponding phylogeny as a resource for future research into the evolution and diversification of biogenic amine receptors.
%DIF -------
Biogenic amine receptors play critical roles in regulating behavior and physiology, particularly within the central nervous system, in both vertebrates and invertebrates. Members of the G-protein coupled receptor (GPCR) family, these receptors interact with endogenous bioamine ligands, such as dopamine, serotonin, and epinephrine, and they are targeted by a wide array of pharmaceuticals. Despite these receptors' clear clinical and biological importance, their evolutionary history remains poorly characterized. In particular, the relationships among biogenic amine receptors and any specific evolutionary constraints acting within distinct receptor subtypes are largely unknown. To advance and facilitate studies in this receptor family, we have constructed a comprehensive, high-quality sequence alignment of vertebrate biogenic amine receptors. In particular, we have integrated a traditional multiple sequence approach with robust structural domain predictions to ensure that alignment columns accurately capture the highly-conserved GPCR structural domains, and we demonstrate how ignoring structural information produces spurious inferences of homology. Using this alignment, we have further constructed a structurally-partitioned maximum-likelihood phylogeny, from which we deduce novel biogenic amine receptor relationships and uncover previously unrecognized lineage-specific receptor clades. Moreover, we find that roughly 1\% of the 3039 sequences in our final alignment are either misannotated or unclassified, and we propose updated classifications for these receptors. We release our comprehensive alignment and its corresponding phylogeny as a resource for future research into the evolution and diversification of biogenic amine receptors. %DIF > 
%DIF -------
\end{abstract}
%DIF PREAMBLE EXTENSION ADDED BY LATEXDIFF
%DIF UNDERLINE PREAMBLE %DIF PREAMBLE
\RequirePackage[normalem]{ulem} %DIF PREAMBLE
\RequirePackage{color}\definecolor{RED}{rgb}{1,0,0}\definecolor{BLUE}{rgb}{0,0,1} %DIF PREAMBLE
\providecommand{\DIFaddtex}[1]{{\protect\color{blue}\uwave{#1}}} %DIF PREAMBLE
\providecommand{\DIFdeltex}[1]{{\protect\color{red}\sout{#1}}}                      %DIF PREAMBLE
%DIF SAFE PREAMBLE %DIF PREAMBLE
\providecommand{\DIFaddbegin}{} %DIF PREAMBLE
\providecommand{\DIFaddend}{} %DIF PREAMBLE
\providecommand{\DIFdelbegin}{} %DIF PREAMBLE
\providecommand{\DIFdelend}{} %DIF PREAMBLE
%DIF FLOATSAFE PREAMBLE %DIF PREAMBLE
\providecommand{\DIFaddFL}[1]{\DIFadd{#1}} %DIF PREAMBLE
\providecommand{\DIFdelFL}[1]{\DIFdel{#1}} %DIF PREAMBLE
\providecommand{\DIFaddbeginFL}{} %DIF PREAMBLE
\providecommand{\DIFaddendFL}{} %DIF PREAMBLE
\providecommand{\DIFdelbeginFL}{} %DIF PREAMBLE
\providecommand{\DIFdelendFL}{} %DIF PREAMBLE
%DIF END PREAMBLE EXTENSION ADDED BY LATEXDIFF
%DIF PREAMBLE EXTENSION ADDED BY LATEXDIFF
%DIF HYPERREF PREAMBLE %DIF PREAMBLE
\providecommand{\DIFadd}[1]{\texorpdfstring{\DIFaddtex{#1}}{#1}} %DIF PREAMBLE
\providecommand{\DIFdel}[1]{\texorpdfstring{\DIFdeltex{#1}}{}} %DIF PREAMBLE
%DIF END PREAMBLE EXTENSION ADDED BY LATEXDIFF

\begin{document}

\flushbottom
\maketitle
\thispagestyle{empty}


\section*{Introduction}

Biogenic amines, such as the molecules serotonin and dopamine, play critical roles in virtually all Metazoans and exert significant influence on both behavior and physiology. In vertebrates, the biogenic amine receptor family, which includes dopamine (DRD), histamine (HRH), trace (TAAR), adrenergic (ADR), muscarinic cholinergic (mAChR), and most serotonin (5HTR) receptors, primarily mediates biogenic amine activity.  Biogenic amine receptors belong to the broad family of G~protein-coupled receptors (GPCRs), one of the largest and most diverse eukaryotic receptor families. Indeed, due to the extensive diversity of biological functions they direct and the ongoing expansion of their ligand repertoire, GPCRs are considered one of the most evolutionarily innovative and successful gene families \citep{BockaertPin1999,Lagerstrom2008}.

Biogenic amine receptors form a clade within the large Rhodopsin-like GPCR family \citep{Fredrikssonetal2003,KakaralaJamil2014}, whose emergence likely accompanied that of the Opisthokont (Fungi and Metazoa) lineage \citep{Krishnan2012}. The Rhodopsin-like family expanded substantially in Metazoa, and the specific diversification of biogenic amine receptors has contributed significantly to central nervous system functioning \citep{Callieretal2003,Nichols2008}. Like all GPCRs, biogenic amine receptors have a \DIFdelbegin \DIFdel{charcterisitic}\DIFdelend \DIFaddbegin \DIFadd{characteristic}\DIFaddend , highly-conserved structure of seven transmembrane (TM) domains separated by three extracellular (ECL) and three intracellular (ICL) loops, and they propagate intracellular signaling through a \DIFdelbegin \DIFdel{G~protein-mediated }\DIFdelend \DIFaddbegin \DIFadd{G-protein-mediated }\DIFaddend pathway. Moreover, these receptors are prominent targets for a wide range \DIFaddbegin \DIFadd{of }\DIFaddend pharmaceuticals aimed to treat myriad diseases such as schizophrenia, migraines, hypertension, allergies and asthma, and stomach ulcers \citep{Schoneberg2004,Eversetal2005,Masonetal2012}.

In spite of these receptors' biological and clinical importance, studies on their evolution are limited and have predominantly focused on individual receptor subtypes, namely TAAR \citep{Gloriametal2005,Lindemann2005,Hashiguchi2007}, DRD \citep{Callieretal2003,Yamamotoetal2013}, and 5HTR \citep{Anbazhagan2010}. Moreover, many of these studies, and indeed studies on the general evolution of the Rhodopsin-like family, have examined very narrow species distributions, for instance specifically teleosts \citep{Gloriametal2005}, primates \citep{Anbazhagan2010}, humans and mice \citep{Vassilatis2003,KakaralaJamil2014}, or even strictly humans \citep{Fredrikssonetal2003}. Thus, virtually no studies \DIFdelbegin \DIFdel{accounting }\DIFdelend \DIFaddbegin \DIFadd{have been conducted that account }\DIFaddend for the full breadth of vertebrate \DIFdelbegin \DIFdel{bioamine receptor sequenceshave been conducted}\DIFdelend \DIFaddbegin \DIFadd{biogenic amine receptor sequences}\DIFaddend .

To gain a comprehensive understanding of this receptor family's evolution, a high-quality multiple sequence alignment (MSA) is needed. MSAs provide the foundation for nearly all comparative sequence analyses, and they are commonly used to locate conserved sequence motifs, identify functionally important residues, and investigate evolutionary histories. As constructing an MSA represents the first step in any sequence analysis, MSA errors are known to bias these downstream analyses \citep{Ogden2006, Wong2008, Jordan2012}. It is therefore crucial to ensure accuracy in MSAs to the extent possible. 

For GPCR sequences, in particular, any MSA should recapitulate the canonical seven-TM structure, which a naive alignment of sequences cannot necessarily accomplish. \DIFdelbegin \DIFdel{While there are certain }\DIFdelend \DIFaddbegin \DIFadd{Several varieties of }\DIFaddend MSA software platforms \DIFdelbegin \DIFdel{that explicitly }\DIFdelend \DIFaddbegin \DIFadd{have been developed that }\DIFaddend incorporate structural information into the alignment algorithm \DIFdelbegin %DIFDELCMD < [%%%
\DIFdel{eg. }%DIFDELCMD < \ %%%
\DIFdel{3DCoffee \mbox{%DIFAUXCMD
\citep{3dcoffee}
}%DIFAUXCMD
, PROMALS3D \mbox{%DIFAUXCMD
\citep{promals3d}
}%DIFAUXCMD
, and HMMALIGN \mbox{%DIFAUXCMD
\citep{hmmer}
}%DIFAUXCMD
}%DIFDELCMD < ]%%%
\DIFdel{, }\DIFdelend \DIFaddbegin \DIFadd{by aligning sequences to a given protein crystal structure \mbox{%DIFAUXCMD
\citep{promals3d, 3dcoffee}
}%DIFAUXCMD
or hidden Markov model (HMM) profile \mbox{%DIFAUXCMD
\citep{hmmer, Chang2012, Hill2012}
}%DIFAUXCMD
. In fact, some programs, such as MP-T \mbox{%DIFAUXCMD
\citep{Hill2012}
}%DIFAUXCMD
and TM-Coffee \mbox{%DIFAUXCMD
\citep{Chang2012}
}%DIFAUXCMD
, cater specifically to membrane proteins. However, all of }\DIFaddend these programs are \DIFdelbegin \DIFdel{fairly }\DIFdelend \DIFaddbegin \DIFadd{extremely }\DIFaddend computationally-intensive and thus ill-suited for large-scale applications. Furthermore, \DIFdelbegin \DIFdel{such }\DIFdelend \DIFaddbegin \DIFadd{many of these }\DIFaddend programs require the use of a single crystal structure or HMM profile to guide sequence alignment. While all GPCRs contain seven TM domains, different GPCR subfamilies, particularly the biogenic amine receptors, feature a wide variety of ICL and ECL sizes. For example, human HRH1 and DRD3 contain roughly 27 and 117 residues, respectively, in their ECL3 domains, and roughly 68 and 14 residues, respectively, in their ICL3 domains [as predicted by GPCRHMM \citep{Wistrand2006}]. Thus, aligning diverse sequences using a single structure may not effectively capture the domain variability across biogenic amine receptor subtypes. \DIFdelbegin \DIFdel{A }\DIFdelend \DIFaddbegin \DIFadd{Instead, a }\DIFaddend desirable alignment strategy would \DIFdelbegin \DIFdel{instead }\DIFdelend anchor all sequences by their conserved TM domains without inappropriately constraining the heterogeneous ECL and ICL domains.

\DIFdelbegin \DIFdel{Here, we have constructed such an MSA using a novel iterative strategy to ensure that all seven TM domains aligned correctly across all sequences, thus yielding the most }\DIFdelend \DIFaddbegin \DIFadd{To this end, we integrated a traditional progressive alignment approach with robust structural predictions to generate a structurally-informed, }\DIFaddend comprehensive (3039 sequences) \DIFdelbegin \DIFdel{and structurally-curated }\DIFdelend \DIFaddbegin \DIFadd{MSA of }\DIFaddend vertebrate biogenic amine \DIFdelbegin \DIFdel{receptor MSA }\DIFdelend \DIFaddbegin \DIFadd{receptors, representing the most extensive such dataset }\DIFaddend to date. \DIFdelbegin \DIFdel{We used this MSA to construct a }\DIFdelend \DIFaddbegin \DIFadd{Through this strategy, we overcame the computational limitations imposed by explicitly structurally-aware MSA construction platforms while still incorporating structural information into MSA construction. We used our structurally-informed MSA to build a }\DIFaddend maximum likelihood (ML) phylogeny of vertebrate biogenic amine receptors, and we found that \DIFdelbegin \DIFdel{our structurally-curated MSA offered dramatic improvements in }\DIFdelend \DIFaddbegin \DIFadd{a partitioned phylogeny which separately considered TM and extramembrane (EM) domains dramatically improved }\DIFaddend phylogenetic fit relative to \DIFdelbegin \DIFdel{a structurally-naive MSA}\DIFdelend \DIFaddbegin \DIFadd{an unpartitioned phylogeny}\DIFaddend . Using this \DIFdelbegin \DIFdel{structurally-aware }\DIFdelend \DIFaddbegin \DIFadd{structurally-partitioned }\DIFaddend phylogeny, we were able to discern relationships among biogenic amine receptor subtypes with a far increased level of sensitivity relative to previous studies, as well as identify novel lineage-specific receptor clades and clarify NCBI annotations for over 30 sequences.

We present this vertebrate biogenic amine receptor MSA and its corresponding phylogeny as a resource for any group interested in studying the dynamic evolutionary processes \DIFdelbegin \DIFdel{, structural , }\DIFdelend \DIFaddbegin \DIFadd{and structural }\DIFaddend and/or functional constraints operating within this class of GPCRs. All data, including MSAs, phylogenies, and sequence descriptions, as well as all code used to generate these data, are freely available from \DIFdelbegin %DIFDELCMD < \\%%%
\texttt{\DIFdel{https://github.com/sjspielman/amine}%DIFDELCMD < \_receptors%%%
}%DIFAUXCMD
\DIFdelend \DIFaddbegin \url{https://github.com/sjspielman/amine\_receptors}\DIFaddend . We expect that these data will prove \DIFdelbegin \DIFdel{extremely }\DIFdelend useful for studying both the broad patterns governing biogenic amine receptor sequence evolution and the evolutionary trends specific to certain receptor subtypes. Further, our \DIFdelbegin \DIFdel{curated }\DIFdelend MSA should serve as a helpful resource in the ongoing development of homology models and pharmaceutical therapeutics targeting these receptors \citep{Kristiansen2004,Ishiguro2004,Eversetal2005,Masonetal2012}.




\section*{Results and Discussion}

\subsection*{Constructing a \DIFdelbegin \DIFdel{structurally-curated }\DIFdelend \DIFaddbegin \DIFadd{structurally-informed }\DIFaddend MSA of biogenic amine receptors}

We collected all \DIFdelbegin \DIFdel{sequences using PSI-BLAST }\DIFdelend \DIFaddbegin \DIFadd{biogenic amine receptor sequences }\DIFaddend from the RefSeq database \citep{refseq} \DIFdelbegin \DIFdel{and filtered data to exclude }\DIFdelend \DIFaddbegin \DIFadd{using PSI-BLAST and removed all }\DIFaddend poor-quality sequences \DIFdelbegin \DIFdel{and/or sequences with excessive ambiguities }\DIFdelend (see \emph{Methods} for details). We \DIFdelbegin \DIFdel{additionally retained only sequences that the program GPCRHMM \mbox{%DIFAUXCMD
\citep{Wistrand2006}
}%DIFAUXCMD
could confidentally classify as GPCRs}\DIFdelend \DIFaddbegin \DIFadd{then used the software GPCRHMM \mbox{%DIFAUXCMD
\citep{Wistrand2006}
}%DIFAUXCMD
to identify whether each sequence in our data set was indeed a GPCR. GPCRHMM uses a hidden Markov model approach to identify GPCRs from protein sequence data alone, and features an exceptionally low false positive rate ($\sim$1\%) as well as a 15\% increase in sensitivity relative to other similar structural prediction programs \mbox{%DIFAUXCMD
\citep{Wistrand2006}
}%DIFAUXCMD
. We removed all sequences which GPCRHMM could not robustly classify as a GPCR}\DIFaddend , leaving a \DIFdelbegin \DIFdel{total }\DIFdelend dataset of 3464 \DIFdelbegin \DIFdel{receptor sequencesto align. 
Before aligning sequences, we again used the program GPCRHMM \mbox{%DIFAUXCMD
\citep{Wistrand2006}
}%DIFAUXCMD
to assign every residue in all protein sequences to its respective }\DIFdelend \DIFaddbegin \DIFadd{protein sequences. 
}

\DIFadd{In addition to identifying GPCR sequences, GPCRHMM can also predict transmembrane domain regions for Rhodopsin-like GPCRs with exceptional accuracy \mbox{%DIFAUXCMD
\citep{SpielmanWilke2013}
}%DIFAUXCMD
. As shown in Figure~\ref{pdb_gpcrhmm}, GPCRHMM yields excellent domain predictions for resolved biogenic amine receptor crystal structures and thus serves as a robust proxy for more computationally-intensive structural predictors. We therefore used GPCRHMM to predict the }\DIFaddend structural domain (\DIFdelbegin \DIFdel{extracellular, transmembrane, }\DIFdelend \DIFaddbegin \DIFadd{TM, extracellular, }\DIFaddend or intracellular) \DIFdelbegin \DIFdel{. GPCRHMM uses a hidden markov model approach to predict GPCR domains with high accuracy from a given protein sequence alone. Indeed, previous work has shown that }\DIFdelend \DIFaddbegin \DIFadd{for each residue in these sequences.
}

\DIFadd{To begin, we then aligned all 3464 protein sequences with MAFFT \mbox{%DIFAUXCMD
\citep{mafftv7}
}%DIFAUXCMD
, with default settings. Using }\DIFaddend GPCRHMM's domain predictions\DIFdelbegin \DIFdel{map exceptionally well to resolved Rhodopsin-like GPCR crystal structures \mbox{%DIFAUXCMD
\citep{SpielmanWilke2013}
}%DIFAUXCMD
. }%DIFDELCMD < 

%DIFDELCMD < %%%
\DIFdel{We then built a structurally-curated MSA from these 3464 protein sequences}\DIFdelend \DIFaddbegin \DIFadd{, we determined the consensus structural domain for each MSA column to assess how well the MSA recapitulated the overarching GPCR domain structure (Figure~\ref{domains}A). We found that, although we had already filtered out putatively non-GPCR sequences, several hundred sequences did not align }\DIFaddend according to the \DIFdelbegin \DIFdel{iterative strategy }\DIFdelend \DIFaddbegin \DIFadd{overarching domain structure. Many sequences were shifted out of structural frame, causing TM domains to inappropriately align with loop domains, or vice versa. Moreover, the mere presence of these misaligned sequences in the naive MSA introduced a substantial amount of gaps, often times within a single TM domain (notably TM1 and TM7, as seen in Figure~\ref{domains}A). While gaps are not inherently problematic in MSAs, the strong evolutionary pressures conserving GPCR structure should prevent large indel events from occurring within TM domains. Thus, many of the large gaps in this alignment were likely produced by the presence of confounding sequences in the dataset.
}

\DIFadd{We therefore adopted an iterative strategy integrating progressive sequence alignment with the GPCRHMM structural predictions to systematically cull poorly-aligned sequences. As }\DIFaddend outlined in Figure~\ref{flowchart}\DIFdelbegin \DIFdel{. For each iteration, we }\DIFdelend \DIFaddbegin \DIFadd{, we first }\DIFaddend aligned protein sequences with MAFFT \citep{mafftv7}. Using the residue domain assignments computed with GPCRHMM, we determined the consensus domain for each column in this MSA. Next, we discarded all sequences for which $\geq 5\%$ of residues did not correspond to their respective column's consensus domain. We realigned the remaining sequences with MAFFT and continued in this manner until no more sequences were discarded. Importantly, this strategy did not require any manual data filtering or visual data inspection, thus avoiding any confounding subjectivity in MSA processing. \DIFdelbegin \DIFdel{The final structurally-curated MSA contained a total of }\DIFdelend \DIFaddbegin \DIFadd{Ultimately, this strategy produced a final MSA which indeed captures the conserved GPCR domain structure, as depicted in Figure~\ref{domains}B. This structurally-informed MSA contains }\DIFaddend 3039 sequences \DIFdelbegin \DIFdel{, which were }\DIFdelend broadly distributed across receptor subtypes \DIFaddbegin \DIFadd{(Table~\ref{tab:abbrev_count}) }\DIFaddend and vertebrate taxa (Figure~\ref{taxa_dist}). \DIFdelbegin %DIFDELCMD < 

%DIFDELCMD < %%%
\DIFdel{From this final MSA, we additionally created a masked MSA, in which protein residues which did not conform to their respective consensus domains were replaced with a ``?''. By masking these positions with an ambiguous character, we ensured that each MSA column strictly contained residues belonging to the same structural domain. In total, 2.69\% of all MSA positions were masked for this MSA.
}%DIFDELCMD < 

%DIFDELCMD < %%%
\DIFdel{We found that our structurally-curated MSA featured far less error than did a structurally-naive MSA, built in MAFFT \mbox{%DIFAUXCMD
\citep{mafftv7}
}%DIFAUXCMD
from all 3464 GPCR sequences without undergoing the process in Figure~\ref{flowchart}. A graphical representation of the naive and structurally-curated MSA, focused specifically on the MSA regions containing the TM domains, is shown in Figure~\ref{domains}. Clearly, many sequences in the naive MSA do not conform to the overarching GPCR structure. Instead, many of these sequences' domains are shifted out of structural frame, leading TM domains to inappropriately align with loop domains, or vice versa, ultimately preventing MSA columns from being truly homologous. Moreover, the mere presence of these misaligned sequences in the naive MSA introduced a substantial amount of gaps, often times within a single TM domain (notably TM1 }\DIFdelend \DIFaddbegin \DIFadd{We additionally noted that the structurally-informed MSA contains canonical Rhodopsin-family GPCR motifs, including the E/DRY motif at the boundary of TM3 }\DIFaddend and \DIFaddbegin \DIFadd{ICL2 (MSA columns 2351-3) and the NPxxY motif at the boundary of }\DIFaddend TM7 \DIFdelbegin \DIFdel{, as seen in Figure~\ref{domains}). By removing such misaligned sequences through the strategy outlined in Figure~\ref{flowchart}, many of these gaps subsequently disppeared, leading to the alignment integrity of individual TM domains. Indeed, the naive MSA contained roughly 23\% more columns than did the structurally-curated MSA, and the vast majority of gaps in the structurally-curated MSAs occurred within the highly heterogenous N-term, C-term, and ICL3 domains}\DIFdelend \DIFaddbegin \DIFadd{and the C-terminal loop (MSA columns 3411-5)}\DIFaddend .



\subsection*{Structurally-aware MSA strongly improves phylogenetic \DIFdelbegin \DIFdel{inference}\DIFdelend \DIFaddbegin \DIFadd{fit}\DIFaddend }

\DIFdelbegin \DIFdel{To demonstrate the utility of analyzing biogenic amine receptors with a structurally-curated MSA, we constructed five distinct }\DIFdelend \DIFaddbegin \DIFadd{Next, we used this structurally-informed MSA to infer a }\DIFaddend maximum likelihood (ML) \DIFdelbegin \DIFdel{phylogenies }\DIFdelend \DIFaddbegin \DIFadd{phylogeny of vertebrate biogenic amine receptors }\DIFaddend in RAxML. \DIFdelbegin \DIFdel{We built a phylogeny using the naive MSA, and we built two phylogenies each from the structural unmasked and masked MSAs. }\DIFdelend Previous work has shown that combined structural and functional constraints impose differing selection pressures in TM vs.\ \DIFdelbegin \DIFdel{extramembrane (EM ) }\DIFdelend \DIFaddbegin \DIFadd{EM }\DIFaddend domains, in turn producing distinct amino-acid \DIFdelbegin \DIFdel{frequency distributions }\DIFdelend \DIFaddbegin \DIFadd{frequencies and evolutionary rates }\DIFaddend in each domain class \citep{Tourasse2000,Stevens2001,Julenius2006,Oberai2009,SpielmanWilke2013,FranzosaXueXia2013}. As our \DIFdelbegin \DIFdel{structurally-curated MSAs }\DIFdelend \DIFaddbegin \DIFadd{MSA }\DIFaddend allowed us to \DIFdelbegin \DIFdel{precisely }\DIFdelend \DIFaddbegin \DIFadd{confidently }\DIFaddend identify each MSA column as either TM or EM, we were able to conduct \DIFdelbegin \DIFdel{far more rigorous phylogentic }\DIFdelend \DIFaddbegin \DIFadd{a more rigorous phylogenetic }\DIFaddend inference using a partitioned analysis. Therefore, \DIFdelbegin \DIFdel{for each of our two structurally-curated MSAs (masked and unmasked), }\DIFdelend we inferred two ML phylogenies: one with two partitions representing TM and EM columns\DIFaddbegin \DIFadd{, respectively, }\DIFaddend and one with a single partition for the entire MSA. \DIFdelbegin %DIFDELCMD < 

%DIFDELCMD < %%%
\DIFdel{As assessed by AIC scores, the structurally-curated masked MSA yielded a far superior phylogeny compared to all other MSAs (Table~\ref{tab:phylo_AIC}), highlighting the benefits of analyzing GPCRs in a structurally-aware context. This finding also underscores that any structurally-aware study must be undertaken cautiously. While partitioning the MSA based on structural domains was clearly beneficial, ensuring that each MSA column strictly contained }\DIFdelend \DIFaddbegin \DIFadd{The former scheme allowed each partition to have unique distributions of evolutionary rate heterogeneity and stationary amino-acid frequencies, thus accounting for the distinct selective regimes in each domain. To ensure as much as possible that the EM and TM partitions contained only }\DIFaddend residues belonging to \DIFdelbegin \DIFdel{the same domainwas critical. Having even a few TM residues in a column assigned to the EM partition, or vice versa, strongly hindered phylogenetic fit}\DIFdelend \DIFaddbegin \DIFadd{their respective structural domain, we created a masked MSA, in which protein residues which did not conform to their respective consensus domains were replaced with a ``?''. All phylogenetic analyses were conducted using this masked MSA}\DIFaddend .

\DIFdelbegin \DIFdel{That the masked structural MSA produced a phylogeny with far better fit than did the unmasked structural MSA additionally reveals the potential benefits to MSA filtering, which entails removing putatively unreliable positions and /or columns from MSAs. In spite of a large body of literature investigating the utility of MSA filtering, the particular circumstances under which filtering is beneficial remain ambiguous \mbox{%DIFAUXCMD
\citep{Castresana2000,Talavera2007,Schloss2010,Penn2010,Jordan2012,Privman2012,Wu2012,SpielmanDawsonWilke2014}
}%DIFAUXCMD
. Here, we find that masking residues based on biological information regarding protein structure can greatly improve }\DIFdelend \DIFaddbegin \DIFadd{We performed 100 ML tree inferences for the unpartitioned and partitioned case each, and we compared the best resulting phylogeny from each scheme using the likelihood ratio test. We found that the partitioned model offered dramatic improvements in }\DIFaddend phylogenetic fit (\DIFdelbegin \DIFdel{Table~\ref{tab:phylo_AIC}), indicating that residue filtering, at least in this circumstance, successfully increased phylogenetic signal. However, we caution that standard MSA filtering algorithms may not always lead to such improvements. For instance, the underlying problem with our structurally-naive MSA was likely not misaligned residues or columns, but rather the presence of confounding sequences which would not align properly with the overarching GPCR structure. Masking particular MSA regions would not have addressed this issue, as confounding sequences would have remained in the MSA even after filters were applied.
Therefore, while MSA filtering may be beneficial, residue masking alone cannot fix a poorly-constructed MSA. 
}\DIFdelend \DIFaddbegin \DIFadd{$p < 1^{-100}$), highlighting the benefits of analyzing GPCRs in a structurally-aware context. We thus proceeded to analyze the partitioned phylogeny more in depth.
}\DIFaddend 


\subsection*{Structurally-aware phylogeny reveals unknown biogenic amine receptor relationships and clades}

Our resulting phylogeny, shown in Figure~\ref{phylogeny}, represents the most comprehensive \DIFdelbegin \DIFdel{and curated }\DIFdelend vertebrate biogenic amine receptor phylogeny to date. This tree broadly captures many known features of biogenic amine receptor evolution, in particular that these receptors do not cluster based on ligand-binding but rather have undergone extensive functional convergent evolution. Indeed, our phylogeny reveals that only two ligand-based receptor classes, mAChR and TAAR, are truly monophyletic. 

Our phylogeny features remarkably high bootstrap support for each distinct clade of receptor subtypes. We additionally find very strong support for three deeper nodes in the phylogeny that reveal the relationships among distinct receptor subtypes. The first contains the three clades HRH1, mAChR, and HRH-3,4, the second contains the clades 5HTR-1, 5HTR-5, and 5HTR-7, and the third contains the 5HTR-4 and TAAR clades. Previous studies have yielded conflicting phylogenetic placements for the 5HTR-7 clade; some have argued that 5HTR-7 is phylogenetically distinct from all other 5HTR sequences \citep{KakaralaJamil2014}, while others have found evidence for a single clade containing 5HTR-5,7 as a sister taxa to a clade containing ADRA1 sequences \citep{Fredrikssonetal2003}. Alternatively, we find moderate-to-strong support for the 5HTR-7 clade having originated before subsequent diversification into 5HTR-5 and 5HTR-1, and we find full support showing that ADRA1 forms an entirely distinct monophyletic group outside all other vertebrate biogenic amine receptors. \DIFdelbegin \DIFdel{In addition, as previously mentioned, our phylogeny reveals }\DIFdelend \DIFaddbegin \DIFadd{We additionally found }\DIFaddend that HRH-3,4 is actually a single monophyletic group. While the HRH-4 clade contains \DIFdelbegin \DIFdel{strictly }\DIFdelend \DIFaddbegin \DIFadd{only }\DIFaddend mammalian sequences, including monotreme (platypus) sequences, HRH-3 sequences are broadly distributed across vertebrate taxa. \DIFdelbegin \DIFdel{We therefore hypothesize }\DIFdelend \DIFaddbegin \DIFadd{This taxonomic distribution suggests }\DIFaddend that HRH-4 \DIFaddbegin \DIFadd{is a mammalian-specific histamine receptor class that }\DIFaddend arose from an HRH-3 duplication concurrent with mammalian origins.

In addition, among the 3039 sequences in \DIFdelbegin \DIFdel{structurally-curated }\DIFdelend \DIFaddbegin \DIFadd{the structurally-informed }\DIFaddend MSA, we identified 31 sequences (\DIFdelbegin \DIFdel{~}\DIFdelend \DIFaddbegin \DIFadd{$\sim$}\DIFaddend 1\% \DIFaddbegin \DIFadd{of our dataset}\DIFaddend ) that we considered misannotated (Table~\ref{tab:classif}), either because the NCBI annotation did not match the sequences' phylogenetic placement or the sequences did not cluster with known biogenic amine receptor types. Several NCBI annotations identified the correct receptor class but the incorrect receptor subtype, whereas other sequences were entirely uncharacterized. \DIFdelbegin \DIFdel{We additionally uncovered }\DIFdelend \DIFaddbegin \DIFadd{In particular, we identified }\DIFaddend an entirely unknown clade of biogenic amine receptors. This unknown clade, \DIFaddbegin \DIFadd{which appears as }\DIFaddend sister to HRH2 \DIFaddbegin \DIFadd{in Figure~\ref{phylogeny}}\DIFaddend , only contains avian sequences and a single \emph{Xenopus tropicalis} (western-clawed frog) sequence. \DIFdelbegin \DIFdel{Two evolutionary scenarios may explain this taxonomic distribution: either this }\DIFdelend \DIFaddbegin \DIFadd{Thus, it is likely that this }\DIFaddend clade emerged concurrently with tetrapods and was secondarily lost in reptiles/birds and mammals\DIFdelbegin \DIFdel{, or this clade represents an avian-specific diversification which the }\emph{\DIFdel{Xenopus tropicalis}} %DIFAUXCMD
\DIFdel{sequence resembles only convergently}\DIFdelend . Interestingly, all but one of this clade's sequences were annotated in NCBI as either octopamine or No9-like receptors, both of which are insect-specific biogenic amine receptors that do not occur in vertebrate taxa \citep{Roeder2005}. The last sequence, alternatively, was annotated as 5HTR-7-like. Taken together, these sequence misannotations suggest an intriguing hypothesis that this clade evolved from an ancestral 5HTR sequence, and subsequent convergent evolution to insect-specific biogenic amine receptors has allowed these receptors to interact with atypical ligands for vertebrates.


\subsection*{Dynamic \DIFaddbegin \DIFadd{lineage-specific }\DIFaddend evolution of the trace-amine associated receptors}
Of particular interest in our phylogeny are the unique evolutionary patterns revealed within the TAAR clade. While all TAAR sequences do cluster together, the \DIFdelbegin \DIFdel{relationships among TAAR subtypes are highly dynamic, reflecting }\DIFdelend \DIFaddbegin \DIFadd{TAAR phylogeny reveals }\DIFaddend the extensive expansion and contraction events characterizing this receptor family's evolution \citep{Lindemann2005,Hashiguchi2007,Staubert2010,Staubert2013}. In fact, the TAAR subtree, displayed in Figure~\ref{taar_tree}, differs somewhat from previously proposed TAAR phylogenies \citep{Lindemann2005, Hashiguchi2007} \DIFdelbegin \DIFdel{. In particular, }\DIFdelend \DIFaddbegin \DIFadd{and reveals }\DIFaddend the presence of \DIFdelbegin \DIFdel{several }\DIFdelend \DIFaddbegin \DIFadd{many }\DIFaddend lineage-specific subclades\DIFdelbegin \DIFdel{as well as unresolved subclades generate novel hypotheses regarding TAAR subtype origins.
While TAAR-2, -3, }\DIFdelend \DIFaddbegin \DIFadd{.
}

\DIFadd{Interestingly, only a single clade in the TAAR phylogeny, comprised of the two subclades TAAR-4 and TAAR-12, contains representatives from the full species distribution considered in our study. Further, the close phylogenetic relationship between these two subclades suggests that TAAR-4 }\DIFaddend and \DIFdelbegin \DIFdel{-4 form a well-resolved monophyly, its sister clade that contains the subtypes TAAR-5, -6, -7, -8, and -9 is less straightforward to interpret. Indeed, the TAAR subtypes -6}\DIFdelend \DIFaddbegin \DIFadd{TAAR-12 are in fact orthologous TAAR subtypes}\DIFaddend , \DIFaddbegin \DIFadd{and the distinct naming for these subtypes apparently emerged simply because TAAR-12 is ray-finned specific. By contrast to this group, all other clades contain distinct, limited species distributions, indicative of repeated gain and loss events. In particular, the broad clade containing subtypes TAAR-6, }\DIFaddend -7, \DIFaddbegin \DIFadd{-8, }\DIFaddend and -9 \DIFdelbegin \DIFdel{do not constitute distinct monophyletic groups, suggesting either poor NCBI sequence annotation or rampant diversification within this subclade. If we assume that these NCBI annotations are reasonably correct, we can deduce that this clade's ancestral sequence was most similar to TAAR-7 and subsequently diversified independently into TAAR-9 and }\DIFdelend \DIFaddbegin \DIFadd{appears to have undergone substantial lineage-specific evolution, with certain subclades only present in marsupial and placental mammals and others only present in lizards and turtles. Moreover, we identified two small clades within the broad }\DIFaddend TAAR-6,\DIFdelbegin \DIFdel{which in turn gave rise to the monophyletic TAAR-8}\DIFdelend \DIFaddbegin \DIFadd{-7,-8,-9 clade that apparent indicate lineage-specific expansions specifically within bovids (labeled ``B" in Figure~\ref{taar_tree}) and rodents (labeled ``R'' in Figure~\ref{taar_tree}), respectively}\DIFaddend .

\DIFdelbegin \DIFdel{Furthermore, }\DIFdelend \DIFaddbegin \DIFadd{Throughout the TAAR phylogeny, ray-finned and lobe-finned fish sequences frequently appear as outgroups to tetrapod-specific clades, indicating progressive diversification tracking large-scale speciation events. However, we do note that }\DIFaddend several lobe-finned fish (coelacanth) sequences are scattered across the TAAR tree and do not clearly cluster with any TAAR subtypes, likely reflecting this lineage's ancient divergence and unique evolutionary trajectory \citep{coelacanth2013}. \DIFdelbegin \DIFdel{The phylogenetic distribution of lobe-finned fish sequences may aid future endeavors to tease apart evolutionary origins of certain TAAR subtypes, specifically whether they represent teleost-specific duplications \mbox{%DIFAUXCMD
\citep{Gloriametal2005}
}%DIFAUXCMD
or whether they represent ancient TAARs that emerged before teleost divergence but were secondarily lost in lobe-finned fish and /or tetrapods}\DIFdelend \DIFaddbegin \DIFadd{Moreover, amphibian sequences are notably absent from this phylogeny, relative to other taxonomic groups. While absence of such sequences in our data set does not necessarily imply that these genes have actually been lost in amphibians, such a hypothesis would be consistent with the overarching gain and loss patterns that TAAR sequences display and thus may merit further study}\DIFaddend .

\DIFdelbegin \DIFdel{In addition, }\DIFdelend \DIFaddbegin \DIFadd{We additionally identified }\DIFaddend a small clade sister to TAAR (labeled in Figures \ref{phylogeny} and \ref{taar_tree} as TAAR$^\ast$) \DIFdelbegin \DIFdel{strictly }\DIFdelend \DIFaddbegin \DIFadd{that only }\DIFaddend contains sequences annotated by NCBI as ``5HTR-4-like.'' At first glance, these annotations might suggest that 5HTR-4 is in fact paraphyletic, diversifying gradually before giving rise to TAARs. However, as all sequences in TAAR$^\ast$ belong taxonomically either to teleost or \emph{Xenopus tropicalis}, we suspect that this clade actually corresponds to the so-called TAAR-V cluster identified by \cite{Hashiguchi2007}. Indeed, the TAAR-V cluster contains a similar taxonomic distribution to our TAAR$^\ast$ and constitutes an outgroup to all other vertebrate TAAR sequences, as our phylogeny similarly displays. 

\subsection*{Phylogenetic methods alone do not suffice to infer the evolutionary history of biogenic amine receptors}
Although we were able to identify several new features of biogenic amine receptor evolution, the majority of deeper splits in the phylogeny had very low bootstrap support, meaning that most of the broader relationships among biogenic amine receptors remain unresolved. This result highlights that a strictly phylogenetic approach cannot fully elucidate the complex evolutionary histories of expanding gene families. In particular, modern phylogenetic methods focus solely on the \DIFdelbegin \DIFdel{substutition }\DIFdelend \DIFaddbegin \DIFadd{substitution }\DIFaddend process and treat MSA gaps simply as missing data. However, gaps actually represent \DIFdelbegin \DIFdel{the evolutionary events of insertion and deletions (indels)}\DIFdelend \DIFaddbegin \DIFadd{insertion and deletion evolutionary events}\DIFaddend , and some have suggested that ignoring this information ultimately hinders phylogenetic accuracy \citep{Morrison2008,Loytynoja2008,Warnow2012,Luanetal2013}. 

This limitation is especially problematic for GPCRs. Following duplication events, GPCRs \DIFdelbegin \DIFdel{appear to }\DIFdelend experience major indel events in their ICL and/or ECL domains, leading to dramatic shifts in loop domain sizes during the sub/neofunctionalization process. Unfortunately, the evolutionary intermediates that existed during these domain transitions have long-since disappeared from genomes, and there is no obvious way to infer the sequences of these missing links. Although the substitution process is key for understanding GPCR evolution, fully \DIFdelbegin \DIFdel{classifiying }\DIFdelend \DIFaddbegin \DIFadd{classifying }\DIFaddend relationships among GPCR families requires some understanding of how these radical domain changes occur. Therefore, additional approaches, such as syntenic analyses \citep{Sundstrom2010,Widmark2011,YegorovGood2012,Hwangetal2013}, combined with the phylogeny presented here should prove useful towards resolving the complete evolutionary history of vertebrate biogenic amine receptors. 


\section*{Conclusions}

We have established a comprehensive, high-quality, \DIFdelbegin \DIFdel{structurally-curated }\DIFdelend \DIFaddbegin \DIFadd{structurally-informed }\DIFaddend MSA of vertebrate biogenic amine receptors. We hope that this MSA, along with its ML phylogeny, will serve as a robust resource for future studies investigating \DIFdelbegin \DIFdel{the evolutionary dynamicsas well as }\DIFdelend \DIFaddbegin \DIFadd{these receptors evolutionary dynamics, }\DIFaddend structural/functional constraints operating within distinct receptor clades\DIFdelbegin \DIFdel{or indeed universal patterns that generally }\DIFdelend \DIFaddbegin \DIFadd{, or overarching patterns that }\DIFaddend govern biogenic amine receptor evolution. Future work may seek to combine the analyses we have \DIFdelbegin \DIFdel{accomplished }\DIFdelend \DIFaddbegin \DIFadd{performed }\DIFaddend here with syntenic or molecular clock \DIFdelbegin \DIFdel{approaches }\DIFdelend \DIFaddbegin \DIFadd{analyses }\DIFaddend to elucidate receptors' origin and precise evolutionary trajectories. Moreover, our \DIFdelbegin \DIFdel{curated }\DIFdelend MSA should prove useful in increasing accuracy in homology modeling and/or pharmaceutical development for these \DIFdelbegin \DIFdel{clincally }\DIFdelend \DIFaddbegin \DIFadd{clinically }\DIFaddend important receptors \citep{Kristiansen2004,Ishiguro2004,Eversetal2005,Masonetal2012}.




\section*{Methods}

\subsection*{Sequence Collection and Processing}
We collected protein sequences using PSI-BLAST \citep{psiblast}, specifically from the RefSeq (v2.2.29+) database \citep{refseq}, for 42 distinct human biogenic amine \DIFdelbegin \DIFdel{receptor }\DIFdelend \DIFaddbegin \DIFadd{protein }\DIFaddend sequences representing the full range of known \DIFaddbegin \DIFadd{such }\DIFaddend receptors in the human genome. To obtain distant yet well-supported orthologs, we ran each PSI-BLAST search for 5 iterations with an e-value cutoff of $10^{-20}$, a sequence identity threshold of 25\%, and a length difference of $\pm50$\% relative to the seed sequence. After combining all sequences recovered from the individual PSI-BLAST searches, we discarded duplicate sequences, leaving a total of 4232 PSI-BLAST results. We then filtered this sequence set to remove sequences from non-vertebrate taxa, sequences annotated as low-quality, pseudogene, and/or partial, and sequences which contained more than 1\% ambiguous residues (i.e.\ B, X, or Z). We additionally used the program GPCRHMM \citep{Wistrand2006} to determine whether a given sequence was indeed a GPCR. We discarded sequences which had either a local or global GPCRHMM score less than 10, both extremely conservative thresholds. Thus, while it is possible that some true GPCRs were discarded, these stringent thresholds for both local and global scores provide high confidence that all retained sequences were indeed GPCRs. Together, these filters left a total of 3464 receptor sequences.


\subsection*{Sequence Alignment and Phylogenetic Reconstruction}
Before aligning sequences, we used the program GPCRHMM \citep{Wistrand2006} to assign each residue in all protein sequences to its respective structural domain (extracellular, transmembrane, or intracellular) using a $0.5$ posterior probability cutoff. We then aligned and filtered sequences according to the strategy outlined in Figure~\ref{flowchart}, which specifically employed MAFFT v7.149b \DIFaddbegin \DIFadd{using the default algorithm }\DIFaddend \citep{mafftv7}. 

All phylogenies were created using RAxML v8.1.1 \citep{raxml} using the LG \DIFdelbegin \DIFdel{+F }\DIFdelend \citep{LG} amino acid \DIFdelbegin \DIFdel{exchangability }\DIFdelend \DIFaddbegin \DIFadd{exchangeability }\DIFaddend matrix with empirical amino acid frequencies \DIFaddbegin \DIFadd{(+F) }\DIFaddend and the CAT model of site heterogeneity \citep{Stamatakis2006}, with the default 25 rate categories. For inferences incorporating structural partitions, we assigned each partition a unique evolutionary model using these settings\DIFaddbegin \DIFadd{, thus allowing each partition to have distinct equilibrium amino-acid frequency and rate distributions}\DIFaddend . Final parameter values for all phylogenetic inferences were optimized with the GAMMA model of heterogeneity. We performed \DIFaddbegin \DIFadd{100 phylogenetic inferences for each parameterization to thoroughly search the tree space, and we used a likelihood ratio test (LRT) to compare the best resulting trees from each parameterization. We computed }\DIFaddend 200 \DIFdelbegin \DIFdel{bootstrap replicates for each }\DIFdelend \DIFaddbegin \DIFadd{bootstrapped trees using RAxML for each resulting }\DIFaddend phylogeny.


\section*{Acknowledgments}
This work was supported in part by NIH grant R01 GM088344, ARO grant W911NF-12-1-0390, DTRA grant HDTRA1-12-C-0007, and NSF Cooperative Agreement No. DBI-0939454 (BEACON Center).  Computational resources were provided by the University of Texas at Austin's Center for Computational Biology and Bioinformatics (CCBB). We would like to thank Ahmad R. Sedaghat, MD, PhD for suggesting biogenic amine receptor evolution as a worthwhile study system.

\DIFdelbegin %DIFDELCMD < \bibliography{bibliography}
%DIFDELCMD < %%%
\DIFdelend \DIFaddbegin \newpage
\DIFaddend 

\bibliography{bibliography}


\DIFaddend \newpage


\section*{Figures and Tables}


\vspace*{5cm}


\begin{table}[htbp]
	\centering
	\begin{tabular}{l l l}
		\hline\noalign{\smallskip}
		\multicolumn{1}{c}{Receptor Class} & \multicolumn{1}{c}{Abbreviation} & \multicolumn{1}{c}{N} \\
		\hline\noalign{\smallskip}
		Serotonin receptors & \quad 5HTR & 972  \\
		\DIFdelbeginFL \DIFdelFL{Trace amine-associated }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{Adrenergic }\DIFaddendFL receptors & \quad \DIFdelbeginFL \DIFdelFL{TAAR }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{ADR }\DIFaddendFL & \DIFdelbeginFL \DIFdelFL{343 }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{611  }\DIFaddendFL \\
		\DIFdelbeginFL \DIFdelFL{Histamine }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{Dopamine }\DIFaddendFL receptors & \quad \DIFdelbeginFL \DIFdelFL{HRH }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{DRD }\DIFaddendFL & \DIFdelbeginFL \DIFdelFL{286 }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{464 }\DIFaddendFL \\
		Muscarinic cholinergic receptors & \quad mAChR & 353  \\
		\DIFdelbeginFL \DIFdelFL{Adrenergic }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{Trace amine-associated }\DIFaddendFL receptors & \quad \DIFdelbeginFL \DIFdelFL{ADR }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{TAAR }\DIFaddendFL & \DIFdelbeginFL \DIFdelFL{611  }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{343 }\DIFaddendFL \\
		\DIFdelbeginFL \DIFdelFL{Dopamine }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{Histamine }\DIFaddendFL receptors & \quad \DIFdelbeginFL \DIFdelFL{DRD }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{HRH }\DIFaddendFL & \DIFdelbeginFL \DIFdelFL{464 }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{286 }\DIFaddendFL \\
		Unknown receptors & \quad Unknown & 10 \\
		\noalign{\smallskip}\hline\noalign{\smallskip} 
	\end{tabular}
	\caption{\label{tab:abbrev_count} Biogenic amine receptor classes, and their abbreviations, considered in this study. The receptor class ``Unknown'' refers to the corresponding uncharacterized clade in Figure~\ref{phylogeny}, and the column ``N'' indicates the total number of sequences for each broad receptor class in our \DIFdelbeginFL \DIFdelFL{structurally-curated }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{structurally-informed }\DIFaddendFL MSA.}
\end{table}




\DIFdelbegin %DIFDELCMD < \vspace*{3cm}
%DIFDELCMD < 

%DIFDELCMD < \begin{table}[htbp]
%DIFDELCMD < 	\centering
%DIFDELCMD < 	\begin{tabular}{l c l l c}
%DIFDELCMD < 		\hline\noalign{\smallskip}
%DIFDELCMD < 		\multicolumn{1}{c}{MSA} & \multicolumn{1}{c}{Partitioned} & \multicolumn{1}{c}{$\ln L$} & \multicolumn{1}{c}{k} & \multicolumn{1}{l}{$\Delta$AIC} \\
%DIFDELCMD < 		\hline\noalign{\smallskip}
%DIFDELCMD < 		%%%
\DIFdel{Structural Masked }%DIFDELCMD < & %%%
\DIFdel{Yes }%DIFDELCMD < & %%%
\DIFdel{-505500.8 }%DIFDELCMD < & %%%
\DIFdel{6115 }%DIFDELCMD < & %%%
\DIFdel{0 }%DIFDELCMD < \\
%DIFDELCMD < 		%%%
\DIFdel{Structural Masked }%DIFDELCMD < & %%%
\DIFdel{No }%DIFDELCMD < & %%%
\DIFdel{-515991.7 }%DIFDELCMD < & %%%
\DIFdel{6095 }%DIFDELCMD < & %%%
\DIFdel{1752 }%DIFDELCMD < \\  
%DIFDELCMD < 		%%%
\DIFdel{Structural Unmasked }%DIFDELCMD < & %%%
\DIFdel{Yes }%DIFDELCMD < & %%%
\DIFdel{-515343.6 }%DIFDELCMD < & %%%
\DIFdel{6115 }%DIFDELCMD < & %%%
\DIFdel{19685 }%DIFDELCMD < \\
%DIFDELCMD < 		%%%
\DIFdel{Structural Unmasked }%DIFDELCMD < & %%%
\DIFdel{No }%DIFDELCMD < & %%%
\DIFdel{-515991.7 }%DIFDELCMD < & %%%
\DIFdel{6095 }%DIFDELCMD < & %%%
\DIFdel{20941 }%DIFDELCMD < \\ 
%DIFDELCMD < 		%%%
\DIFdel{Naive }%DIFDELCMD < & %%%
\DIFdel{No }%DIFDELCMD < &  %%%
\DIFdel{-589703.7 }%DIFDELCMD < & %%%
\DIFdel{6945 }%DIFDELCMD < & %%%
\DIFdel{170047 }%DIFDELCMD < \\
%DIFDELCMD < 		\noalign{\smallskip}\hline\noalign{\smallskip} 
%DIFDELCMD < 	\end{tabular}
%DIFDELCMD < 	%%%
%DIFDELCMD < \caption{%
{%DIFAUXCMD
%DIFDELCMD < \label{tab:phylo_AIC} %%%
\DIFdel{$\Delta$AIC scores relative to the best performing for phylogenies. The column labeled ``Partitioned'' indicates whether phylogenetic inference was conducted with distinct TM (transmembrane) and EM (extramembrane) partitions. AIC is computed as AIC $= 2(k - \ln L)$, where $k$ is the number of free parameters of the model, and $\ln L$ is the log-likelihood \mbox{%DIFAUXCMD
\citep{Akaike1974,BurnhamAnderson2004}
}%DIFAUXCMD
. AIC scores are reported here relative to the phylogeny with the lowest AIC score (structural masked with partitions), representing the best-fitting phylogeny.}}
%DIFAUXCMD
%DIFDELCMD < \end{table}
%DIFDELCMD < 

%DIFDELCMD < %%%
\DIFdelend \newpage

\begin{table}[htbp]
	\centering
	\begin{tabular}{l l l l l l}
		\hline\noalign{\smallskip}
		\multicolumn{1}{c}{Protein ID} & \multicolumn{1}{c}{Nucleotide ID} & \multicolumn{2}{c}{Current Classification} & \multicolumn{2}{c}{Proposed Classification} \\
		\hline\noalign{\smallskip}
		XP\_005797918.1 & XM\_005797861.1 & \qquad \qquad & DRD-2 & \qquad \qquad & DRD-3 \\
		XP\_003967971.1 & XM\_003967922.1 & \qquad \qquad & DRD-2 & \qquad \qquad & DRD-3 \\
		NP\_001266433.1 & NM\_001279504.1 & \qquad \qquad & mAChR-4 & \qquad \qquad & mAChR-2 \\
		XP\_001520508.2 & XM\_001520458.3 & \qquad \qquad & HRH-3 & \qquad \qquad & HRH-4 \\
		XP\_005282846.1 & XM\_005282789.1 & \qquad \qquad & HRH-4 & \qquad \qquad & HRH-3 \\
		XP\_001920844.1 & XM\_001920809.1 & \qquad \qquad & TAAR-4-like & \qquad \qquad & TAAR-12 \\
		NP\_001076571.1 & NM\_001083102.1 & \qquad \qquad & TAAR-64 & \qquad \qquad & TAAR-13 \\
		XP\_006014096.1 & XM\_006014034.1 & \qquad \qquad & TAAR-9-like & \qquad \qquad & TAAR-4 \\
		XP\_003201718.2 & XM\_003201670.2 & \qquad \qquad & TAAR-1-like & \qquad \qquad & TAAR-10 \\
		NP\_001076546.1 & NM\_001083077.1 & \qquad \qquad & TAAR-11-like & \qquad \qquad & TAAR-10 \\
		NP\_001083418.1 & NM\_001089949.1 & \qquad \qquad & uncharacterized & \qquad \qquad & ADRB \\
		NP\_001103208.1 & NM\_001109738.1 & \qquad \qquad & uncharacterized & \qquad \qquad & HRH-2 \\
		NP\_001124143.1 & NM\_001130671.1 & \qquad \qquad & uncharacterized & \qquad \qquad & TAAR-12 \\
		XP\_001337671.1 & XM\_001337635.2 & \qquad \qquad & 5HTR-4-like & \qquad \qquad & TAAR$^\ast$ \\
		XP\_003976403.1 & XM\_003976354.1 & \qquad \qquad & 5HTR-4-like & \qquad \qquad & TAAR$^\ast$ \\
		XP\_005810466.1 & XM\_005810409.1 & \qquad \qquad & 5HTR-4-like & \qquad \qquad & TAAR$^\ast$ \\
		XP\_003454279.1 & XM\_003454231.1 & \qquad \qquad & 5HTR-4-like & \qquad \qquad & TAAR$^\ast$ \\
		XP\_004549625.1 & XM\_004549568.1 & \qquad \qquad & 5HTR-4-like & \qquad \qquad & TAAR$^\ast$ \\
		XP\_002935532.2 & XM\_002935486.2 & \qquad \qquad & 5HTR-4-like & \qquad \qquad & TAAR$^\ast$ \\
		XP\_006013317.1 & XM\_006013255.1 & \qquad \qquad & 5HTR-4-like & \qquad \qquad & TAAR$^\ast$ \\
		XP\_005510029.1 & XM\_005509972.1 & \qquad \qquad & 5HTR-7-like & \qquad \qquad & Unknown \\
		XP\_002187301.2 & XM\_002187265.2 & \qquad \qquad & Octopamine receptor-like & \qquad \qquad & Unknown \\
		XP\_002937327.2 & XM\_002937281.2 & \qquad \qquad & Octopamine receptor-like & \qquad \qquad & Unknown \\
		XP\_005045681.1 & XM\_005045624.1 & \qquad \qquad & Octopamine receptor-like & \qquad \qquad & Unknown \\
		XP\_005144673.1 & XM\_005144616.1 & \qquad \qquad & Octopamine receptor-like & \qquad \qquad & Unknown \\
		XP\_005229932.1 & XM\_005229875.1 & \qquad \qquad & Octopamine receptor-like & \qquad \qquad & Unknown \\
		XP\_005428400.1 & XM\_005428343.1 & \qquad \qquad & Probable GPCR No9-like & \qquad \qquad & Unknown \\
		XP\_005490920.1 & XM\_005490863.1 & \qquad \qquad & Probable GPCR No9-like & \qquad \qquad & Unknown \\
		XP\_005518128.1 & XM\_005518071.1 & \qquad \qquad & Probable GPCR No9-like & \qquad \qquad & Unknown \\
		XP\_006111669.1 & XM\_006111607.1 & \qquad \qquad & Octopamine receptor-like & \qquad \qquad & Unknown \\
		XP\_420867.2 & XM\_420867.4 & \qquad \qquad & Octopamine receptor & \qquad \qquad & Unknown \\
		\noalign{\smallskip}\hline\noalign{\smallskip} 
	\end{tabular}
	\caption{\label{tab:classif} Misannotated and uncharacterized sequences identified through phylogenetic analysis. Based on sequence placement in the \DIFdelbeginFL \DIFdelFL{structurally-curated }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{structurally-partitioned }\DIFaddendFL phylogeny\DIFdelbeginFL \DIFdelFL{(Figure~\ref{phylogeny}) }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{, }\DIFaddendFL we propose updated classifications for 31 biogenic amine receptor sequences. The proposed classifications ``Unknown'' and ``TAAR$^\ast$'' refer to the \DIFdelbeginFL \DIFdelFL{corresponding }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{correspondingly-labeled }\DIFaddendFL clades in Figure~\ref{phylogeny}.}
\end{table}

\newpage

\begin{figure}[htbp]
	\DIFaddbeginFL \centerline{\includegraphics[width=15cm]{../figures/pdb_gpcrhmm.pdf}}
	\caption{\label{pdb_gpcrhmm} \DIFaddFL{GPCRHMM domain predictions for representative biogenic amine receptor crystal structures from the Protein Data Bank (PDB). Gene names are shown in capital letters above each structure, and corresponding PDB IDs are shown in parentheses. Dark blue represents predicted extracellular residues, red represents predicted TM residues, light blue represents predicted intracellular residues.}}
\end{figure}



\vspace*{2cm}

\begin{figure}[htbp]
	\centerline{\includegraphics[width=8in]{../figures/domains_naive_struc.png}}
	\caption{\label{domains} Graphical representation of a subset of the A) \DIFdelbeginFL \DIFdelFL{naive }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{structurally-naive }\DIFaddendFL and B) \DIFdelbeginFL \DIFdelFL{structurally-curated }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{structurally-informed }\DIFaddendFL biogenic amine receptor MSAs. Each \DIFdelbeginFL \DIFdelFL{image }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{panel }\DIFaddendFL displays 130 MSA rows focused specifically on the MSA section containing the seven TM domains. Dark blue represents predicted extracellular residues, red represents predicted TM residues, lighter blue represents predicted intracellular residues, and gray represents MSA gaps. The bottom bar below each MSA figure shows the consensus domain structure for each MSA. The \DIFdelbeginFL \DIFdelFL{naive }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{structurally-naive }\DIFaddendFL MSA was built all 3464 putative GPCR sequences in MAFFT, whereas the structurally-informed MSA was built using the iterative strategy outlined in Figure~\ref{flowchart}. Note that all columns which contain only gaps in this MSA subset have been removed from this figure for visual clarity.}
\end{figure}


\newpage

\begin{figure}[htbp]
	\DIFaddendFL \centerline{\includegraphics[width=18cm]{../figures/alignment_flowchart.pdf}}
	\caption{\label{flowchart} Iterative alignment strategy \DIFaddbeginFL \DIFaddFL{used }\DIFaddendFL to \DIFdelbeginFL \DIFdelFL{create a structurally-curated MSA of }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{generate the structurally-informed }\DIFaddendFL vertebrate biogenic amine \DIFdelbeginFL \DIFdelFL{receptors}\DIFdelendFL \DIFaddbeginFL \DIFaddFL{receptor MSA}\DIFaddendFL . A total of 3464 sequences were initially input (``Validated GPCR protein sequences''), and the final \DIFaddbeginFL \DIFaddFL{structurally-informed }\DIFaddendFL MSA contained 3039 protein sequences. Residues marked with ``I'' represent intracellular residues, those marked with ``M'' represent \DIFdelbeginFL \DIFdelFL{transmembrane }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{TM }\DIFaddendFL residues, and those marked with ``E'' represent extracellular residues. MSA gaps were treated as missing data when determining each column's consensus structural domain. Sequenced were removed (``remove highly discordant sequences'') if $\geq 5\%$ of columns belonged to a different structural domain than the respective consensus domain. \DIFdelbeginFL %DIFDELCMD < }
}
\end{figure}


\DIFdelbegin %DIFDELCMD < \vspace*{4cm}
%DIFDELCMD < 

%DIFDELCMD < %%%
\DIFdelend \begin{figure}[htbp]
 \DIFaddbeginFL \centerline{\includegraphics[width=7cm]{../figures/taxonomic_distribution.pdf}}
	\caption{\label{taxa_dist} Cladogram of the taxonomic distribution of all sequences in the final \DIFdelbeginFL \DIFdelFL{structurally-curated }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{structurally-informed }\DIFaddendFL MSA. All sequences belonged to the Euteleostomi clade of jawed vertebrates. Numbers in parentheses indicate the total number of sequences from the respective clade. We note that our MSA is particularly enriched for sequences from Eutherian (placental mammal) species, likely due to the stringent filters we applied to sequence collection that favored fully-sequenced genomes.}
\DIFaddendFL \end{figure}


\newpage

\begin{figure}[htbp]
	\DIFdelbeginFL %DIFDELCMD < \centerline{\includegraphics[width=18cm]{figures/masked_part_phylogeny.pdf}}
%DIFDELCMD < 	%%%
\DIFdelendFL \DIFaddbeginFL \centerline{\includegraphics[width=18cm]{../figures/vert_amine_tree.pdf}}
	\DIFaddendFL \caption{\label{phylogeny} \DIFdelbeginFL \DIFdelFL{Maximum likelihood }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{Maximum-likelihood }\DIFaddendFL phylogeny of vertebrate biogenic amine receptors built using the masked \DIFdelbeginFL \DIFdelFL{structural }\DIFdelendFL \DIFaddbeginFL \DIFaddFL{structurally-informed }\DIFaddendFL MSA in RAxML. Nodes with open circles indicate $\geq 50\%$ bootstrap support, and nodes with closed circles \DIFdelbeginFL \DIFdelFL{and thick lines }\DIFdelendFL indicate $\geq 90\%$ bootstrap support. Biogenic amine receptors are abbreviated as in Table~\ref{tab:abbrev_count}. The clade labeled ``Unknown'' could not be clearly identified as one of the major receptor types and may represent a previously unrecognized biogenic amine receptor clade. \DIFaddbeginFL \DIFaddFL{Note that the root shown on this tree has been placed arbitrarily.}\DIFaddendFL }
\end{figure}


\newpage

\begin{figure}[htbp]
	\DIFdelbeginFL %DIFDELCMD < \centerline{\includegraphics[width=15cm]{figures/taar_phylogeny.pdf}}
%DIFDELCMD < 	%%%
\DIFdelendFL \DIFaddbeginFL \centerline{\includegraphics[width=15cm]{../figures/taar_taxonomy.pdf}}
	\DIFaddendFL \caption{\label{taar_tree} Subclade of the TAAR receptors within the phylogeny shown in Figure~\ref{phylogeny}. Nodes with open circles indicate $\geq 50\%$ bootstrap support, and nodes with closed circles \DIFdelbeginFL \DIFdelFL{and thick lines }\DIFdelendFL indicate $\geq 90\%$ bootstrap support\DIFaddbeginFL \DIFaddFL{. The subclades within the TAAR-6,7,8,9 clade labeled as ``B'' and ``R'' indicate clades containing only bovid and rodent sequences, respectively}\DIFaddendFL .}
\end{figure}




\end{document}
